{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PZQ resistance: genome-wide association study\n",
    "\n",
    "\n",
    "## Aim\n",
    "\n",
    "This notebook describes the different analyses related to mapping the QTL of praziquantel (PZQ) resistance. We perform a genome with association study on a population of *Schistosoma mansoni* (SmLE-PZQ-R) which is polymorphic for PZQ resistance ([Couto *et al.*](https://doi.org/10.1590/s0074-02762011000200006)). The population was treated with a dose of 24 Âµg/mL of PZQ and survival of the worms was objectively assessed using a lactacte assay as a proxy. Pools of the 20% low and 20% high lactacte producer were made. The GWAS was conducted with two independent biological replicate and each library was sequenced in triplicate for assessing allele frequency.\n",
    "\n",
    "After identifying the QTL related to resistance, we performed a marker assisted selection to create two populations: an enriched population in resistance allele (ER) and an enriched population in sensitive allele (ES).\n",
    "\n",
    "\n",
    "\n",
    "## Environment and data\n",
    "\n",
    "Installation of the environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2021-01-15T17:54:58.863Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n"
     ]
    }
   ],
   "source": [
    "conda env create -f .env/env.yml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The cell below must be run each time a new Jupyter session is run."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-05T21:48:28.120309Z",
     "start_time": "2021-06-05T21:48:24.350124Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Activate the environment\n",
    "source $(sed \"s,/bin/conda,,\" <<<$CONDA_EXE)/etc/profile.d/conda.sh\n",
    "conda activate PZQ-R_DNA\n",
    "\n",
    "# Remove potential variable interferences\n",
    "export PERL5LIB=\"\"\n",
    "export PYTHONNOUSERSITE=1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The cell below must be run only once at the time of the environment creation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Installing needed R packages\n",
    "Rscript \".env/R package dependencies.R\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Phenotypic data\n",
    "\n",
    "The phenotypic data for the F0 parental population, the F1s and F2s is available on [Zenodo](https://doi.org/10.5281/zenodo.4383248). Data is already present in the repository but can be downloaded if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Data directory\n",
    "pdir=\"data/phenotypes\"\n",
    "[[ ! -d \"$pdir\" ]] && mkdir -p \"$pdir\"\n",
    "\n",
    "# Download files\n",
    "wget -q -P \"$pdir\" https://zenodo.org/record/4383248/files/F0_parental_populations.csv\n",
    "wget -q -P \"$pdir\" https://zenodo.org/record/4383248/files/F1.csv\n",
    "wget -q -P \"$pdir\" https://zenodo.org/record/4383248/files/F2.csv\n",
    "wget -q -P \"$pdir\" https://zenodo.org/record/4383248/files/sex.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sequencing data\n",
    "\n",
    "This step downloads the fastq files of the different samples from the SRA repository."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Data directory\n",
    "ldir=\"data/libraries\"\n",
    "[[ ! -d \"$ldir\" ]] && mkdir -p \"$ldir\"\n",
    "\n",
    "# Bioproject\n",
    "bioproject=   ## !! To UPDATE PRJNA667697\n",
    "\n",
    "# Download related information to data project\n",
    "wget -q -O runinfo \"http://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?save=efetch&rettype=runinfo&db=sra&term=${bioproject}\"\n",
    "\n",
    "# Field of interest (library name and weblink)\n",
    "fdn=$(head -n 1 runinfo | tr \",\" \"\\n\" | grep -w -n \"LibraryName\" | cut -d \":\" -f 1)\n",
    "fdr=$(head -n 1 runinfo | tr \",\" \"\\n\" | grep -w -n \"Run\" | cut -d \":\" -f 1)\n",
    "\n",
    "# Download fastq files\n",
    "while read line\n",
    "do\n",
    "    # Filename and download link\n",
    "    fln=$(cut -d \",\" -f $fdn <<<$line)\n",
    "    run=$(cut -d \",\" -f $fdr <<<$line)\n",
    "    \n",
    "    # Download\n",
    "    echo \"$fln\"\n",
    "    [[ ! -d \"$ldir/$fln/\" ]] && mkdir -p \"$ldir/$fln/\"\n",
    "    fastq-dump -O \"$ldir/$fln/\" --split-files \"$run\"\n",
    "    \n",
    "    mv \"$ldir/$fln/${run}_1.fastq\" \"$ldir/$fln/${fln}_R1.fastq\"\n",
    "    mv \"$ldir/$fln/${run}_2.fastq\" \"$ldir/$fln/${fln}_R2.fastq\"\n",
    "        \n",
    "done < <(tail -n +2 runinfo | sed \"/^$/d\")\n",
    "\n",
    "# Compress files\n",
    "pigz \"$ldir/\"*/*\n",
    "\n",
    "rm runinfo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genome data\n",
    "\n",
    "The genome data is downloaded from the [WormBase ParaSite](https://parasite.wormbase.org). We use the data from the version 14 (WBPS14). The data is then indexed for the different tools used."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T22:16:09.973102Z",
     "start_time": "2021-01-27T22:16:09.810094Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "gdir=\"data/genome\"\n",
    "[[ ! -d \"$gdir\" ]] && mkdir -p \"$gdir\"\n",
    "\n",
    "# Download and unzip data\n",
    "wget -P \"$gdir\" ftp://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS14/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa.gz\n",
    "pigz -d \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-15T17:54:47.231632Z",
     "start_time": "2021-01-15T17:54:43.804663Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] fail to open file '/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa' : No such file or directory\n",
      "[fai_build] fail to open the FASTA file /schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n"
     ]
    }
   ],
   "source": [
    "# Preparing indices\n",
    "bwa index \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\"\n",
    "samtools faidx \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\"\n",
    "gatk CreateSequenceDictionary -R \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### snpEff database\n",
    "\n",
    "To annotate the functional consequences of the variants in the VCF file, we use the snpEff tool. Because the reference genome is recent, we have to build a snpEff database in order to annotate the variants. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T23:00:05.337322Z",
     "start_time": "2021-01-27T22:16:22.082712Z"
    }
   },
   "outputs": [],
   "source": [
    "# Create snpEff folder\n",
    "sedir=\"$gdir/snpEff/\"\n",
    "mkdir \"$sedir\"\n",
    "\n",
    "# Download the S. mansoni genome annotation\n",
    "wget -P \"$gdir\" ftp://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS14/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS14.annotations.gff3.gz\n",
    "\n",
    "pigz -d \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.annotations.gff3.gz\"\n",
    "\n",
    "# Convert gff to gtf to make it easier for snpEff\n",
    "gffread \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.annotations.gff3\" -T -o \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.annotations.gtf\"\n",
    "\n",
    "# Custom snpEff config file\n",
    "cp \"$CONDA_PREFIX\"/share/snpeff-4.3.1t-4/snpEff.config \"$sedir\"\n",
    "\n",
    "## Identify mitochondrial chromosome\n",
    "mito=$(grep -i \">.*mito.*\" \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\" | sed -r \"s/>(.*) .*/\\1/\")\n",
    "\n",
    "##  Add schistosome genome and specify mitochondrial chromosome\n",
    "sed -i -r \"s|data.dir = ./data/|data.dir = .| ; s|(# Ebola virus)|# S. mansoni\\nsmansoni.genome: Schistosoma_mansoni\\n\\tsmansoni.$mito.codonTable : Trematode_Mitochondrial\\n\\n\\1|\" \"$sedir\"/snpEff.config\n",
    "\n",
    "# Create the mandatory folder and link to the needed files\n",
    "mkdir \"$sedir\"/genomes\n",
    "ln -s \"$PWD/$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\" \"$PWD/$sedir/genomes/smansoni.fa\"\n",
    "\n",
    "mkdir \"$sedir\"/smansoni\n",
    "ln -s \"$PWD/$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.annotations.gtf\" \"$PWD/$sedir/smansoni/genes.gtf\"\n",
    "\n",
    "# Generate the database\n",
    "snpEff build -v -c \"$sedir/snpEff.config\" smansoni"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Known variants\n",
    "\n",
    "List of variants from the SmLE x SmHR crosses that segregated in a Mendelian fashion in F1 ([Valentim *et al.* 2013](https://doi.org/10.1126/science.1243106)). This list was generated from alignments using the v5 genome. This has been lifted over to the latest genome using the [flo pipeline](https://github.com/wurmlab/flo/tree/727f10b2b1c57a0514835d302d7f6345d3a34ffb)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Index file\n",
    "gatk IndexFeatureFile -I \"$gdir/sm_dbSNP_v7.vcf\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Phenotype\n",
    "\n",
    "ADD WINKA DATA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sequencing data processing\n",
    "\n",
    "### Snakemake pipeline\n",
    "\n",
    "The snakemake pipeline performs the following steps:\n",
    "* alignment of each library against the reference genome\n",
    "* marking duplicates\n",
    "* base quality recalibration\n",
    "* generating alignment statistics\n",
    "* calling variants (gVCF) for each library\n",
    "* combining gVCF of all libraries\n",
    "* genotyping all samples per contig\n",
    "* merging all the contigs to generate final VCF file\n",
    "\n",
    "The VCF file produced will then be used to identify possible QTLs link to the cercarial production."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2021-01-18T01:25:35.273Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
      "\u001b[33mProvided cluster nodes: 384\u001b[0m\n",
      "\u001b[33mJob counts:\n",
      "\tcount\tjobs\n",
      "\t12\talignment\n",
      "\t1\tall\n",
      "\t12\tbsqr\n",
      "\t12\tcalling\n",
      "\t2\tcombining\n",
      "\t320\tgenotype_variants\n",
      "\t12\tindexing1\n",
      "\t12\tindexing2\n",
      "\t12\tindexing3\n",
      "\t12\tmark_duplicates\n",
      "\t1\tmerge_variants\n",
      "\t12\tstats\n",
      "\t420\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:51 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_2_WGS/SmLE-PZQ-R_Exp1_Recovered_2_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_2_WGS/SmLE-PZQ-R_Exp1_Recovered_2_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_2_WGS/SmLE-PZQ-R_Exp1_Recovered_2_WGS_sorted.bam\n",
      "    jobid: 3\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 3 with external jobid 'Your job 28328 (\"snakejob.alignment.3.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted.bam\n",
      "    jobid: 8\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 8 with external jobid 'Your job 28329 (\"snakejob.alignment.8.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted.bam\n",
      "    jobid: 1\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 1 with external jobid 'Your job 28330 (\"snakejob.alignment.1.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted.bam\n",
      "    jobid: 12\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 12 with external jobid 'Your job 28331 (\"snakejob.alignment.12.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted.bam\n",
      "    jobid: 6\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 6 with external jobid 'Your job 28332 (\"snakejob.alignment.6.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted.bam\n",
      "    jobid: 10\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 10 with external jobid 'Your job 28333 (\"snakejob.alignment.10.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted.bam\n",
      "    jobid: 9\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 9 with external jobid 'Your job 28334 (\"snakejob.alignment.9.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted.bam\n",
      "    jobid: 11\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 11 with external jobid 'Your job 28335 (\"snakejob.alignment.11.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted.bam\n",
      "    jobid: 2\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 2 with external jobid 'Your job 28336 (\"snakejob.alignment.2.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:52 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted.bam\n",
      "    jobid: 7\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 7 with external jobid 'Your job 28337 (\"snakejob.alignment.7.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:53 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted.bam\n",
      "    jobid: 4\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 4 with external jobid 'Your job 28338 (\"snakejob.alignment.4.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sun Jan 17 19:25:53 2021]\u001b[0m\n",
      "\u001b[32mrule alignment:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_R1.fastq.gz, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_R2.fastq.gz, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted.bam\n",
      "    jobid: 5\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 5 with external jobid 'Your job 28339 (\"snakejob.alignment.5.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 07:51:38 2021]\u001b[0m\n",
      "\u001b[32mFinished job 6.\u001b[0m\n",
      "\u001b[32m1 of 420 steps (0.24%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 07:51:38 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted.bam\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted.bam.bai\n",
      "    jobid: 18\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 18 with external jobid 'Your job 28550 (\"snakejob.indexing1.18.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:05:36 2021]\u001b[0m\n",
      "\u001b[32mFinished job 18.\u001b[0m\n",
      "\u001b[32m2 of 420 steps (0.48%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:05:36 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.log\n",
      "    jobid: 30\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 30 with external jobid 'Your job 28551 (\"snakejob.mark_duplicates.30.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:42:41 2021]\u001b[0m\n",
      "\u001b[32mFinished job 12.\u001b[0m\n",
      "\u001b[32m3 of 420 steps (0.71%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:42:41 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted.bam.bai\n",
      "    jobid: 24\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:42:41 2021]\u001b[0m\n",
      "\u001b[32mFinished job 7.\u001b[0m\n",
      "\u001b[32m4 of 420 steps (0.95%) done\u001b[0m\n",
      "\u001b[33mSubmitted job 24 with external jobid 'Your job 28552 (\"snakejob.indexing1.24.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:42:41 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted.bam.bai\n",
      "    jobid: 19\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 19 with external jobid 'Your job 28553 (\"snakejob.indexing1.19.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:47:49 2021]\u001b[0m\n",
      "\u001b[32mFinished job 19.\u001b[0m\n",
      "\u001b[32m5 of 420 steps (1%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:47:49 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.log\n",
      "    jobid: 31\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 31 with external jobid 'Your job 28554 (\"snakejob.mark_duplicates.31.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:58:28 2021]\u001b[0m\n",
      "\u001b[32mFinished job 24.\u001b[0m\n",
      "\u001b[32m6 of 420 steps (1%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 08:58:28 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.log\n",
      "    jobid: 36\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 36 with external jobid 'Your job 28555 (\"snakejob.mark_duplicates.36.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:00:19 2021]\u001b[0m\n",
      "\u001b[32mFinished job 30.\u001b[0m\n",
      "\u001b[32m7 of 420 steps (2%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:00:19 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.bam.bai\n",
      "    jobid: 42\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 42 with external jobid 'Your job 28556 (\"snakejob.indexing2.42.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:18:41 2021]\u001b[0m\n",
      "\u001b[32mFinished job 42.\u001b[0m\n",
      "\u001b[32m8 of 420 steps (2%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:18:41 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam\n",
      "    jobid: 54\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 54 with external jobid 'Your job 28557 (\"snakejob.bsqr.54.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:18:52 2021]\u001b[0m\n",
      "\u001b[32mFinished job 31.\u001b[0m\n",
      "\u001b[32m9 of 420 steps (2%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:18:52 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.bam.bai\n",
      "    jobid: 43\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 43 with external jobid 'Your job 28558 (\"snakejob.indexing2.43.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:26:12 2021]\u001b[0m\n",
      "\u001b[32mFinished job 43.\u001b[0m\n",
      "\u001b[32m10 of 420 steps (2%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:26:12 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam\n",
      "    jobid: 55\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 55 with external jobid 'Your job 28559 (\"snakejob.bsqr.55.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:53:23 2021]\u001b[0m\n",
      "\u001b[32mFinished job 36.\u001b[0m\n",
      "\u001b[32m11 of 420 steps (3%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 09:53:23 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.bam.bai\n",
      "    jobid: 48\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 48 with external jobid 'Your job 28560 (\"snakejob.indexing2.48.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:07:21 2021]\u001b[0m\n",
      "\u001b[32mFinished job 11.\u001b[0m\n",
      "\u001b[32m12 of 420 steps (3%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:07:21 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted.bam.bai\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    jobid: 23\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 23 with external jobid 'Your job 28561 (\"snakejob.indexing1.23.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:07:22 2021]\u001b[0m\n",
      "\u001b[32mFinished job 5.\u001b[0m\n",
      "\u001b[32m13 of 420 steps (3%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:07:22 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted.bam.bai\n",
      "    jobid: 17\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 17 with external jobid 'Your job 28562 (\"snakejob.indexing1.17.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:13:58 2021]\u001b[0m\n",
      "\u001b[32mFinished job 48.\u001b[0m\n",
      "\u001b[32m14 of 420 steps (3%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:13:58 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam\n",
      "    jobid: 60\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 60 with external jobid 'Your job 28563 (\"snakejob.bsqr.60.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:18:00 2021]\u001b[0m\n",
      "\u001b[32mFinished job 23.\u001b[0m\n",
      "\u001b[32m15 of 420 steps (4%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:18:00 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.log\n",
      "    jobid: 35\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 35 with external jobid 'Your job 28564 (\"snakejob.mark_duplicates.35.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:18:45 2021]\u001b[0m\n",
      "\u001b[32mFinished job 17.\u001b[0m\n",
      "\u001b[32m16 of 420 steps (4%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:18:45 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.log\n",
      "    jobid: 29\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 29 with external jobid 'Your job 28565 (\"snakejob.mark_duplicates.29.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:22:03 2021]\u001b[0m\n",
      "\u001b[32mFinished job 8.\u001b[0m\n",
      "\u001b[32m17 of 420 steps (4%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:22:03 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted.bam.bai\n",
      "    jobid: 20\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 20 with external jobid 'Your job 28566 (\"snakejob.indexing1.20.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:22:47 2021]\u001b[0m\n",
      "\u001b[32mFinished job 9.\u001b[0m\n",
      "\u001b[32m18 of 420 steps (4%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:22:47 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted.bam.bai\n",
      "    jobid: 21\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 21 with external jobid 'Your job 28567 (\"snakejob.indexing1.21.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:27:55 2021]\u001b[0m\n",
      "\u001b[32mFinished job 20.\u001b[0m\n",
      "\u001b[32m19 of 420 steps (5%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:27:56 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.log\n",
      "    jobid: 32\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 32 with external jobid 'Your job 28568 (\"snakejob.mark_duplicates.32.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:30:52 2021]\u001b[0m\n",
      "\u001b[32mFinished job 21.\u001b[0m\n",
      "\u001b[32m20 of 420 steps (5%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:30:52 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.log\n",
      "    jobid: 33\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 33 with external jobid 'Your job 28569 (\"snakejob.mark_duplicates.33.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:33:48 2021]\u001b[0m\n",
      "\u001b[32mFinished job 55.\u001b[0m\n",
      "\u001b[32m21 of 420 steps (5%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:33:48 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 67\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 67 with external jobid 'Your job 28570 (\"snakejob.indexing3.67.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:44:27 2021]\u001b[0m\n",
      "\u001b[32mFinished job 67.\u001b[0m\n",
      "\u001b[32m22 of 420 steps (5%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:44:27 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS.gvcf.gz\n",
      "    jobid: 91\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 91 with external jobid 'Your job 28571 (\"snakejob.calling.91.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:44:28 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 79\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_3_WGS\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 79 with external jobid 'Your job 28572 (\"snakejob.stats.79.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_3_WGS/SmLE-PZQ-R_Exp2_Contracted_3_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:51:48 2021]\u001b[0m\n",
      "\u001b[32mFinished job 79.\u001b[0m\n",
      "\u001b[32m23 of 420 steps (5%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:56:58 2021]\u001b[0m\n",
      "\u001b[32mFinished job 29.\u001b[0m\n",
      "\u001b[32m24 of 420 steps (6%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:56:58 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.bam.bai\n",
      "    jobid: 41\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 41 with external jobid 'Your job 28573 (\"snakejob.indexing2.41.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:58:03 2021]\u001b[0m\n",
      "\u001b[32mFinished job 35.\u001b[0m\n",
      "\u001b[32m25 of 420 steps (6%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 10:58:03 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.bam.bai\n",
      "    jobid: 47\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 47 with external jobid 'Your job 28574 (\"snakejob.indexing2.47.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:36 2021]\u001b[0m\n",
      "\u001b[32mFinished job 47.\u001b[0m\n",
      "\u001b[32m26 of 420 steps (6%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:36 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam\n",
      "    jobid: 59\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:36 2021]\u001b[0m\n",
      "\u001b[32mFinished job 33.\u001b[0m\n",
      "\u001b[32m27 of 420 steps (6%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:36 2021]\u001b[0m\n",
      "\u001b[32mFinished job 32.\u001b[0m\n",
      "\u001b[32m28 of 420 steps (7%) done\u001b[0m\n",
      "\u001b[33mSubmitted job 59 with external jobid 'Your job 28575 (\"snakejob.bsqr.59.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:36 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.bam.bai\n",
      "    jobid: 45\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 45 with external jobid 'Your job 28576 (\"snakejob.indexing2.45.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:36 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.bam.bai\n",
      "    jobid: 44\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 44 with external jobid 'Your job 28577 (\"snakejob.indexing2.44.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:37 2021]\u001b[0m\n",
      "\u001b[32mFinished job 41.\u001b[0m\n",
      "\u001b[32m29 of 420 steps (7%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:07:37 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam\n",
      "    jobid: 53\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 53 with external jobid 'Your job 28578 (\"snakejob.bsqr.53.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:17:31 2021]\u001b[0m\n",
      "\u001b[32mFinished job 45.\u001b[0m\n",
      "\u001b[32m30 of 420 steps (7%) done\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:17:31 2021]\u001b[0m\n",
      "\u001b[32mFinished job 44.\u001b[0m\n",
      "\u001b[32m31 of 420 steps (7%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:17:31 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam\n",
      "    jobid: 57\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 57 with external jobid 'Your job 28579 (\"snakejob.bsqr.57.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 11:17:31 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam\n",
      "    jobid: 56\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 56 with external jobid 'Your job 28580 (\"snakejob.bsqr.56.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:06:44 2021]\u001b[0m\n",
      "\u001b[32mFinished job 53.\u001b[0m\n",
      "\u001b[32m32 of 420 steps (8%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:06:44 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 65\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 65 with external jobid 'Your job 28581 (\"snakejob.indexing3.65.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:15:34 2021]\u001b[0m\n",
      "\u001b[32mFinished job 65.\u001b[0m\n",
      "\u001b[32m33 of 420 steps (8%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:15:34 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.flagstat\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    jobid: 77\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 77 with external jobid 'Your job 28582 (\"snakejob.stats.77.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:15:34 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS.gvcf.gz\n",
      "    jobid: 89\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 89 with external jobid 'Your job 28583 (\"snakejob.calling.89.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:17:45 2021]\u001b[0m\n",
      "\u001b[32mFinished job 59.\u001b[0m\n",
      "\u001b[32m34 of 420 steps (8%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:17:45 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 71\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 71 with external jobid 'Your job 28584 (\"snakejob.indexing3.71.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_2_WGS/SmLE-PZQ-R_Exp1_Contracted_2_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:18:30 2021]\u001b[0m\n",
      "\u001b[32mFinished job 77.\u001b[0m\n",
      "\u001b[32m35 of 420 steps (8%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:24:44 2021]\u001b[0m\n",
      "\u001b[32mFinished job 54.\u001b[0m\n",
      "\u001b[32m36 of 420 steps (9%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:24:44 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 66\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 66 with external jobid 'Your job 28585 (\"snakejob.indexing3.66.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:29:52 2021]\u001b[0m\n",
      "\u001b[32mFinished job 71.\u001b[0m\n",
      "\u001b[32m37 of 420 steps (9%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:29:52 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 83\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:29:53 2021]\u001b[0m\n",
      "\u001b[32mFinished job 56.\u001b[0m\n",
      "\u001b[32m38 of 420 steps (9%) done\u001b[0m\n",
      "\u001b[33mSubmitted job 83 with external jobid 'Your job 28586 (\"snakejob.stats.83.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:29:53 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS.gvcf.gz\n",
      "    jobid: 95\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 95 with external jobid 'Your job 28587 (\"snakejob.calling.95.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:29:53 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 68\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 68 with external jobid 'Your job 28588 (\"snakejob.indexing3.68.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:31:43 2021]\u001b[0m\n",
      "\u001b[32mFinished job 57.\u001b[0m\n",
      "\u001b[32m39 of 420 steps (9%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:31:43 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 69\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 69 with external jobid 'Your job 28589 (\"snakejob.indexing3.69.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:31:54 2021]\u001b[0m\n",
      "\u001b[32mFinished job 1.\u001b[0m\n",
      "\u001b[32m40 of 420 steps (10%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:31:54 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted.bam.bai\n",
      "    jobid: 13\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 13 with external jobid 'Your job 28590 (\"snakejob.indexing1.13.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:37:03 2021]\u001b[0m\n",
      "\u001b[32mFinished job 2.\u001b[0m\n",
      "\u001b[32m41 of 420 steps (10%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:37:03 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted.bam.bai\n",
      "    jobid: 14\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 14 with external jobid 'Your job 28591 (\"snakejob.indexing1.14.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_1_WGS/SmLE-PZQ-R_Exp1_Recovered_1_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:42:33 2021]\u001b[0m\n",
      "\u001b[32mFinished job 83.\u001b[0m\n",
      "\u001b[32m42 of 420 steps (10%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:46:36 2021]\u001b[0m\n",
      "\u001b[32mFinished job 66.\u001b[0m\n",
      "\u001b[32m43 of 420 steps (10%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:46:36 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS.gvcf.gz\n",
      "    jobid: 90\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 90 with external jobid 'Your job 28592 (\"snakejob.calling.90.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:46:36 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.bam.bai\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 78\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 78 with external jobid 'Your job 28593 (\"snakejob.stats.78.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:46:47 2021]\u001b[0m\n",
      "\u001b[32mFinished job 10.\u001b[0m\n",
      "\u001b[32m44 of 420 steps (10%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:46:47 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted.bam.bai\n",
      "    jobid: 22\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 22 with external jobid 'Your job 28594 (\"snakejob.indexing1.22.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:48:37 2021]\u001b[0m\n",
      "\u001b[32mFinished job 13.\u001b[0m\n",
      "\u001b[32m45 of 420 steps (11%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:48:37 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.log\n",
      "    jobid: 25\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 25 with external jobid 'Your job 28595 (\"snakejob.mark_duplicates.25.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:49:21 2021]\u001b[0m\n",
      "\u001b[32mFinished job 68.\u001b[0m\n",
      "\u001b[32m46 of 420 steps (11%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:49:21 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 80\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 80 with external jobid 'Your job 28596 (\"snakejob.stats.80.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:49:21 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS.gvcf.gz\n",
      "    jobid: 92\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Contracted_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 92 with external jobid 'Your job 28597 (\"snakejob.calling.92.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:51:12 2021]\u001b[0m\n",
      "\u001b[32mFinished job 69.\u001b[0m\n",
      "\u001b[32m47 of 420 steps (11%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:51:12 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS.gvcf.gz\n",
      "    jobid: 93\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 93 with external jobid 'Your job 28598 (\"snakejob.calling.93.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:51:12 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 81\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 81 with external jobid 'Your job 28599 (\"snakejob.stats.81.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:56:21 2021]\u001b[0m\n",
      "\u001b[32mFinished job 14.\u001b[0m\n",
      "\u001b[32m48 of 420 steps (11%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 12:56:21 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.log\n",
      "    jobid: 26\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 26 with external jobid 'Your job 28600 (\"snakejob.mark_duplicates.26.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_3_WGS/SmLE-PZQ-R_Exp1_Contracted_3_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:02:57 2021]\u001b[0m\n",
      "\u001b[32mFinished job 81.\u001b[0m\n",
      "\u001b[32m49 of 420 steps (12%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_1_WGS/SmLE-PZQ-R_Exp2_Contracted_1_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:07:00 2021]\u001b[0m\n",
      "\u001b[32mFinished job 78.\u001b[0m\n",
      "\u001b[32m50 of 420 steps (12%) done\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:07:44 2021]\u001b[0m\n",
      "\u001b[32mFinished job 22.\u001b[0m\n",
      "\u001b[32m51 of 420 steps (12%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:07:44 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.log\n",
      "    jobid: 34\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 34 with external jobid 'Your job 28601 (\"snakejob.mark_duplicates.34.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Contracted_2_WGS/SmLE-PZQ-R_Exp2_Contracted_2_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:08:28 2021]\u001b[0m\n",
      "\u001b[32mFinished job 80.\u001b[0m\n",
      "\u001b[32m52 of 420 steps (12%) done\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:15:49 2021]\u001b[0m\n",
      "\u001b[32mFinished job 4.\u001b[0m\n",
      "\u001b[32m53 of 420 steps (13%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:15:49 2021]\u001b[0m\n",
      "\u001b[32mrule indexing1:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted.bam.bai\n",
      "    jobid: 16\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 16 with external jobid 'Your job 28602 (\"snakejob.indexing1.16.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:24:15 2021]\u001b[0m\n",
      "\u001b[32mFinished job 16.\u001b[0m\n",
      "\u001b[32m54 of 420 steps (13%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:24:15 2021]\u001b[0m\n",
      "\u001b[32mrule mark_duplicates:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    jobid: 28\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 28 with external jobid 'Your job 28603 (\"snakejob.mark_duplicates.28.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:28:18 2021]\u001b[0m\n",
      "\u001b[32mFinished job 60.\u001b[0m\n",
      "\u001b[32m55 of 420 steps (13%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:28:18 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 72\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 72 with external jobid 'Your job 28604 (\"snakejob.indexing3.72.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:30:08 2021]\u001b[0m\n",
      "\u001b[32mFinished job 25.\u001b[0m\n",
      "\u001b[32m56 of 420 steps (13%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:30:08 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.bam.bai\n",
      "    jobid: 37\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 37 with external jobid 'Your job 28605 (\"snakejob.indexing2.37.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:38:35 2021]\u001b[0m\n",
      "\u001b[32mFinished job 37.\u001b[0m\n",
      "\u001b[32m57 of 420 steps (14%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:38:35 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam\n",
      "    jobid: 49\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 49 with external jobid 'Your job 28606 (\"snakejob.bsqr.49.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:53:39 2021]\u001b[0m\n",
      "\u001b[32mFinished job 26.\u001b[0m\n",
      "\u001b[32m58 of 420 steps (14%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:53:39 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.bam.bai\n",
      "    jobid: 38\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 38 with external jobid 'Your job 28607 (\"snakejob.indexing2.38.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:54:11 2021]\u001b[0m\n",
      "\u001b[32mFinished job 72.\u001b[0m\n",
      "\u001b[32m59 of 420 steps (14%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:54:11 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS.gvcf.gz\n",
      "    jobid: 96\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:54:11 2021]\u001b[0m\n",
      "\u001b[32mFinished job 34.\u001b[0m\n",
      "\u001b[32m60 of 420 steps (14%) done\u001b[0m\n",
      "\u001b[33mSubmitted job 96 with external jobid 'Your job 28608 (\"snakejob.calling.96.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:54:11 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 84\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 84 with external jobid 'Your job 28609 (\"snakejob.stats.84.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 13:54:11 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.bam.bai\n",
      "    jobid: 46\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 46 with external jobid 'Your job 28610 (\"snakejob.indexing2.46.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.log.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 14:07:06 2021]\u001b[0m\n",
      "\u001b[32mFinished job 28.\u001b[0m\n",
      "\u001b[32m61 of 420 steps (15%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 14:07:06 2021]\u001b[0m\n",
      "\u001b[32mrule indexing2:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.bam.bai\n",
      "    jobid: 40\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 40 with external jobid 'Your job 28847 (\"snakejob.indexing2.40.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:20:28 2021]\u001b[0m\n",
      "\u001b[32mFinished job 40.\u001b[0m\n",
      "\u001b[32m62 of 420 steps (15%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:20:28 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam\n",
      "    jobid: 52\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 52 with external jobid 'Your job 29514 (\"snakejob.bsqr.52.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_3_WGS/SmLE-PZQ-R_Exp2_Recovered_3_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:27:04 2021]\u001b[0m\n",
      "\u001b[32mFinished job 84.\u001b[0m\n",
      "\u001b[32m63 of 420 steps (15%) done\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:28:54 2021]\u001b[0m\n",
      "\u001b[32mFinished job 46.\u001b[0m\n",
      "\u001b[32m64 of 420 steps (15%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:28:54 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam\n",
      "    jobid: 58\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 58 with external jobid 'Your job 29534 (\"snakejob.bsqr.58.sh\") has been submitted'.\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32m[Mon Jan 18 16:28:55 2021]\u001b[0m\n",
      "\u001b[32mFinished job 38.\u001b[0m\n",
      "\u001b[32m65 of 420 steps (15%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:28:55 2021]\u001b[0m\n",
      "\u001b[32mrule bsqr:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD.grp, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam\n",
      "    jobid: 50\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 50 with external jobid 'Your job 29535 (\"snakejob.bsqr.50.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:32:12 2021]\u001b[0m\n",
      "\u001b[32mFinished job 49.\u001b[0m\n",
      "\u001b[32m66 of 420 steps (16%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:32:12 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 61\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 61 with external jobid 'Your job 29543 (\"snakejob.indexing3.61.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:43:57 2021]\u001b[0m\n",
      "\u001b[32mFinished job 61.\u001b[0m\n",
      "\u001b[32m67 of 420 steps (16%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:43:57 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 73\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 73 with external jobid 'Your job 29553 (\"snakejob.stats.73.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:43:58 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS.gvcf.gz\n",
      "    jobid: 85\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Recovered_3_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 85 with external jobid 'Your job 29554 (\"snakejob.calling.85.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Recovered_3_WGS/SmLE-PZQ-R_Exp1_Recovered_3_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 16:46:54 2021]\u001b[0m\n",
      "\u001b[32mFinished job 73.\u001b[0m\n",
      "\u001b[32m68 of 420 steps (16%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:41:37 2021]\u001b[0m\n",
      "\u001b[32mFinished job 58.\u001b[0m\n",
      "\u001b[32m69 of 420 steps (16%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:41:37 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 70\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 70 with external jobid 'Your job 29558 (\"snakejob.indexing3.70.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:45:41 2021]\u001b[0m\n",
      "\u001b[32mFinished job 50.\u001b[0m\n",
      "\u001b[32m70 of 420 steps (17%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:45:41 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 62\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 62 with external jobid 'Your job 29559 (\"snakejob.indexing3.62.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:47:53 2021]\u001b[0m\n",
      "\u001b[32mFinished job 52.\u001b[0m\n",
      "\u001b[32m71 of 420 steps (17%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:47:53 2021]\u001b[0m\n",
      "\u001b[32mrule indexing3:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam.bai\n",
      "    jobid: 64\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 64 with external jobid 'Your job 29560 (\"snakejob.indexing3.64.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:57:47 2021]\u001b[0m\n",
      "\u001b[32mFinished job 70.\u001b[0m\n",
      "\u001b[32m72 of 420 steps (17%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:57:47 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 82\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 82 with external jobid 'Your job 29561 (\"snakejob.stats.82.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:57:47 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS.gvcf.gz\n",
      "    jobid: 94\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 94 with external jobid 'Your job 29562 (\"snakejob.calling.94.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:59:39 2021]\u001b[0m\n",
      "\u001b[32mFinished job 64.\u001b[0m\n",
      "\u001b[32m73 of 420 steps (17%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:59:39 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 76\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 76 with external jobid 'Your job 29563 (\"snakejob.stats.76.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 17:59:39 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS.gvcf.gz\n",
      "    jobid: 88\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp2_Recovered_2_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 88 with external jobid 'Your job 29564 (\"snakejob.calling.88.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam.bai.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 18:00:45 2021]\u001b[0m\n",
      "\u001b[32mFinished job 62.\u001b[0m\n",
      "\u001b[32m74 of 420 steps (18%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 18:00:45 2021]\u001b[0m\n",
      "\u001b[32mrule calling:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam.bai, data/genome/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa, data/genome/sm_dbSNP_v7.vcf\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS.gvcf.gz\n",
      "    jobid: 86\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 86 with external jobid 'Your job 29565 (\"snakejob.calling.86.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 18:00:45 2021]\u001b[0m\n",
      "\u001b[32mrule stats:\n",
      "    input: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam, data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.bam.bai\n",
      "    output: data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.flagstat\n",
      "    jobid: 74\n",
      "    wildcards: sample=SmLE-PZQ-R_Exp1_Contracted_1_WGS\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mSubmitted job 74 with external jobid 'Your job 29566 (\"snakejob.stats.74.sh\") has been submitted'.\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_1_WGS/SmLE-PZQ-R_Exp2_Recovered_1_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 18:02:56 2021]\u001b[0m\n",
      "\u001b[32mFinished job 82.\u001b[0m\n",
      "\u001b[32m75 of 420 steps (18%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp2_Recovered_2_WGS/SmLE-PZQ-R_Exp2_Recovered_2_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 18:10:18 2021]\u001b[0m\n",
      "\u001b[32mFinished job 76.\u001b[0m\n",
      "\u001b[32m76 of 420 steps (18%) done\u001b[0m\n",
      "\u001b[33mWrite-protecting output file data/libraries/1-GWAS/SmLE-PZQ-R_Exp1_Contracted_1_WGS/SmLE-PZQ-R_Exp1_Contracted_1_WGS_sorted_MD_recal.flagstat.\u001b[0m\n",
      "\u001b[32m[Mon Jan 18 18:10:18 2021]\u001b[0m\n",
      "\u001b[32mFinished job 74.\u001b[0m\n",
      "\u001b[32m77 of 420 steps (18%) done\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# Status directory\n",
    "statdir=status\n",
    "[[ ! -d \"$statdir\" ]] && mkdir -p \"$statdir\"\n",
    "\n",
    "snakemake --snakefile GWAS.smk --cluster \"qsub -V -cwd -o $statdir -j y -r y -pe smp 10 -S /bin/bash\" --jobs 384 -w 120"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Variant annotation\n",
    "\n",
    "Functional consequences of variants is annotated to the VCF file using snpEff."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-28T01:31:42.095422Z",
     "start_time": "2021-01-28T01:31:41.886287Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PZQ_GWAS\n",
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "grep -m 1 VCF_PFX GWAS.smk | cut -d \"\\\"\" -f 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-28T01:41:33.794373Z",
     "start_time": "2021-01-28T01:35:01.789821Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) 00:00:00\tSnpEff version SnpEff 4.3t (build 2017-11-24 10:18), by Pablo Cingolani\n",
      "00:00:00\tCommand: 'ann'\n",
      "00:00:00\tReading configuration file 'data/genome/snpEff//snpEff.config'. Genome: 'smansoni'\n",
      "00:00:00\tReading config file: /data/infectious/schistosome/06 - PZQ resistance/1-PZQ-DNA/data/genome/snpEff/snpEff.config\n",
      "00:00:01\tdone\n",
      "00:00:01\tReading database for genome version 'smansoni' from file '/data/infectious/schistosome/06 - PZQ resistance/1-PZQ-DNA/data/genome/snpEff/./smansoni/snpEffectPredictor.bin' (this might take a while)\n",
      "00:00:06\tdone\n",
      "00:00:06\tLoading Motifs and PWMs\n",
      "00:00:06\tBuilding interval forest\n",
      "00:00:06\tGene 'gene:Smp_320850' spans across coordinate zero: Assuming circular chromosome, creating mirror gene at the end.\n",
      "\tGene        :Gene_SM_V7_2H012:-7145-7773\n",
      "\tNew gene    :Gene_SM_V7_2H012:34721-49639\n",
      "\tChrsomosome :Chromosome_SM_V7_2H012:1-41866\n",
      "00:00:06\tTotal: 1 added as circular mirrored genes (appended '_circ' to IDs).\n",
      "00:00:08\tdone.\n",
      "00:00:08\tGenome stats :\n",
      "#-----------------------------------------------\n",
      "# Genome name                : 'Schistosoma_mansoni'\n",
      "# Genome version             : 'smansoni'\n",
      "# Genome ID                  : 'smansoni[0]'\n",
      "# Has protein coding info    : true\n",
      "# Has Tr. Support Level info : true\n",
      "# Genes                      : 10174\n",
      "# Protein coding genes       : 10145\n",
      "#-----------------------------------------------\n",
      "# Transcripts                : 14529\n",
      "# Avg. transcripts per gene  : 1.43\n",
      "# TSL transcripts            : 0\n",
      "#-----------------------------------------------\n",
      "# Checked transcripts        : \n",
      "#               AA sequences :      0 ( 0.00% )\n",
      "#              DNA sequences :      0 ( 0.00% )\n",
      "#-----------------------------------------------\n",
      "# Protein coding transcripts : 14500\n",
      "#              Length errors :      3 ( 0.02% )\n",
      "#  STOP codons in CDS errors :      2 ( 0.01% )\n",
      "#         START codon errors :     43 ( 0.30% )\n",
      "#        STOP codon warnings :      8 ( 0.06% )\n",
      "#              UTR sequences :  14183 ( 97.62% )\n",
      "#               Total Errors :     44 ( 0.30% )\n",
      "#-----------------------------------------------\n",
      "# Cds                        : 113492\n",
      "# Exons                      : 126867\n",
      "# Exons with sequence        : 126867\n",
      "# Exons without sequence     : 0\n",
      "# Avg. exons per transcript  : 8.73\n",
      "#-----------------------------------------------\n",
      "# Number of chromosomes      : 320\n",
      "# Chromosomes                : Format 'chromo_name size codon_table'\n",
      "#\t\t'SM_V7_1'\t88881357\tStandard\n",
      "#\t\t'SM_V7_ZW'\t88385488\tStandard\n",
      "#\t\t'SM_V7_3'\t50458499\tStandard\n",
      "#\t\t'SM_V7_2'\t48130368\tStandard\n",
      "#\t\t'SM_V7_4'\t47279781\tStandard\n",
      "#\t\t'SM_V7_5'\t25256119\tStandard\n",
      "#\t\t'SM_V7_6'\t24989083\tStandard\n",
      "#\t\t'SM_V7_7'\t19288021\tStandard\n",
      "#\t\t'SM_V7_W001'\t935871\tStandard\n",
      "#\t\t'SM_V7_W002'\t793022\tStandard\n",
      "#\t\t'SM_V7_W003'\t513415\tStandard\n",
      "#\t\t'SM_V7_ZWU001'\t289827\tStandard\n",
      "#\t\t'SM_V7_W004'\t279311\tStandard\n",
      "#\t\t'SM_V7_5H001'\t220032\tStandard\n",
      "#\t\t'SM_V7_2H001'\t200280\tStandard\n",
      "#\t\t'SM_V7_ZWU002'\t173519\tStandard\n",
      "#\t\t'SM_V7_2H002'\t165532\tStandard\n",
      "#\t\t'SM_V7_U001'\t160195\tStandard\n",
      "#\t\t'SM_V7_W005'\t149646\tStandard\n",
      "#\t\t'SM_V7_ZWH001'\t146669\tStandard\n",
      "#\t\t'SM_V7_ZWU003'\t146658\tStandard\n",
      "#\t\t'SM_V7_W006'\t144428\tStandard\n",
      "#\t\t'SM_V7_2H003'\t138432\tStandard\n",
      "#\t\t'SM_V7_2H004'\t133822\tStandard\n",
      "#\t\t'SM_V7_2H005'\t133597\tStandard\n",
      "#\t\t'SM_V7_U002'\t131157\tStandard\n",
      "#\t\t'SM_V7_2H006'\t125934\tStandard\n",
      "#\t\t'SM_V7_4H001'\t124601\tStandard\n",
      "#\t\t'SM_V7_W007'\t122025\tStandard\n",
      "#\t\t'SM_V7_7H001'\t110230\tStandard\n",
      "#\t\t'SM_V7_ZWH002'\t103925\tStandard\n",
      "#\t\t'SM_V7_4H002'\t102981\tStandard\n",
      "#\t\t'SM_V7_W009'\t101813\tStandard\n",
      "#\t\t'SM_V7_U003'\t100028\tStandard\n",
      "#\t\t'SM_V7_2H007'\t99744\tStandard\n",
      "#\t\t'SM_V7_6H001'\t96912\tStandard\n",
      "#\t\t'SM_V7_5H003'\t96552\tStandard\n",
      "#\t\t'SM_V7_5H002'\t96260\tStandard\n",
      "#\t\t'SM_V7_7H002'\t95647\tStandard\n",
      "#\t\t'SM_V7_3H001'\t95497\tStandard\n",
      "#\t\t'SM_V7_U004'\t89015\tStandard\n",
      "#\t\t'SM_V7_2H008'\t85875\tStandard\n",
      "#\t\t'SM_V7_W011'\t79719\tStandard\n",
      "#\t\t'SM_V7_6H002'\t77628\tStandard\n",
      "#\t\t'SM_V7_2H009'\t76736\tStandard\n",
      "#\t\t'SM_V7_5H004'\t76371\tStandard\n",
      "#\t\t'SM_V7_5H005'\t73321\tStandard\n",
      "#\t\t'SM_V7_5H006'\t72929\tStandard\n",
      "#\t\t'SM_V7_5H008'\t71870\tStandard\n",
      "#\t\t'SM_V7_5H007'\t71845\tStandard\n",
      "#\t\t'SM_V7_ZWH003'\t66674\tStandard\n",
      "#\t\t'SM_V7_6H003'\t64465\tStandard\n",
      "#\t\t'SM_V7_W008'\t64370\tStandard\n",
      "#\t\t'SM_V7_1H002'\t64117\tStandard\n",
      "#\t\t'SM_V7_1H001'\t64018\tStandard\n",
      "#\t\t'SM_V7_6H004'\t63063\tStandard\n",
      "#\t\t'SM_V7_5H009'\t62792\tStandard\n",
      "#\t\t'SM_V7_ZWH004'\t62045\tStandard\n",
      "#\t\t'SM_V7_4H003'\t61159\tStandard\n",
      "#\t\t'SM_V7_W010'\t61066\tStandard\n",
      "#\t\t'SM_V7_3001'\t60127\tStandard\n",
      "#\t\t'SM_V7_ZWH007'\t59995\tStandard\n",
      "#\t\t'SM_V7_ZWH005'\t59940\tStandard\n",
      "#\t\t'SM_V7_7H003'\t59764\tStandard\n",
      "#\t\t'SM_V7_ZWH006'\t59443\tStandard\n",
      "#\t\t'SM_V7_ZWH008'\t59083\tStandard\n",
      "#\t\t'SM_V7_5H010'\t58595\tStandard\n",
      "#\t\t'SM_V7_ZWH009'\t57825\tStandard\n",
      "#\t\t'SM_V7_1H003'\t56863\tStandard\n",
      "#\t\t'SM_V7_ZWU004'\t56281\tStandard\n",
      "#\t\t'SM_V7_5H011'\t55767\tStandard\n",
      "#\t\t'SM_V7_U005'\t55760\tStandard\n",
      "#\t\t'SM_V7_6H006'\t55696\tStandard\n",
      "#\t\t'SM_V7_W013'\t55224\tStandard\n",
      "#\t\t'SM_V7_2H010'\t55153\tStandard\n",
      "#\t\t'SM_V7_6H007'\t54162\tStandard\n",
      "#\t\t'SM_V7_U006'\t53943\tStandard\n",
      "#\t\t'SM_V7_W014'\t53884\tStandard\n",
      "#\t\t'SM_V7_6H005'\t53532\tStandard\n",
      "#\t\t'SM_V7_7H004'\t53144\tStandard\n",
      "#\t\t'SM_V7_4H004'\t51325\tStandard\n",
      "#\t\t'SM_V7_1H004'\t50418\tStandard\n",
      "#\t\t'SM_V7_6H008'\t49591\tStandard\n",
      "#\t\t'SM_V7_W012'\t49306\tStandard\n",
      "#\t\t'SM_V7_1H005'\t49039\tStandard\n",
      "#\t\t'SM_V7_5H012'\t48833\tStandard\n",
      "#\t\t'SM_V7_2H011'\t48623\tStandard\n",
      "#\t\t'SM_V7_U007'\t48619\tStandard\n",
      "#\t\t'SM_V7_W016'\t48598\tStandard\n",
      "#\t\t'SM_V7_1H007'\t48179\tStandard\n",
      "#\t\t'SM_V7_W019'\t47855\tStandard\n",
      "#\t\t'SM_V7_1H008'\t47782\tStandard\n",
      "#\t\t'SM_V7_1H006'\t47641\tStandard\n",
      "#\t\t'SM_V7_1001'\t47563\tStandard\n",
      "#\t\t'SM_V7_3H002'\t47115\tStandard\n",
      "#\t\t'SM_V7_ZWH010'\t47057\tStandard\n",
      "#\t\t'SM_V7_4H006'\t46612\tStandard\n",
      "#\t\t'SM_V7_5H013'\t46444\tStandard\n",
      "#\t\t'SM_V7_ZWH011'\t46325\tStandard\n",
      "#\t\t'SM_V7_3H003'\t45846\tStandard\n",
      "#\t\t'SM_V7_ZWU005'\t45821\tStandard\n",
      "#\t\t'SM_V7_4H007'\t45807\tStandard\n",
      "#\t\t'SM_V7_U008'\t45745\tStandard\n",
      "#\t\t'SM_V7_W018'\t45515\tStandard\n",
      "#\t\t'SM_V7_ZWH012'\t45489\tStandard\n",
      "#\t\t'SM_V7_6H009'\t45418\tStandard\n",
      "#\t\t'SM_V7_4H005'\t45362\tStandard\n",
      "#\t\t'SM_V7_ZWH013'\t44693\tStandard\n",
      "#\t\t'SM_V7_1H009'\t44509\tStandard\n",
      "#\t\t'SM_V7_1H010'\t44155\tStandard\n",
      "#\t\t'SM_V7_3H004'\t44047\tStandard\n",
      "#\t\t'SM_V7_3H005'\t43874\tStandard\n",
      "#\t\t'SM_V7_6H010'\t43669\tStandard\n",
      "#\t\t'SM_V7_4H008'\t43640\tStandard\n",
      "#\t\t'SM_V7_ZWH014'\t43015\tStandard\n",
      "#\t\t'SM_V7_5H014'\t42658\tStandard\n",
      "#\t\t'SM_V7_ZWH015'\t42645\tStandard\n",
      "#\t\t'SM_V7_1002'\t42467\tStandard\n",
      "#\t\t'SM_V7_4H009'\t42081\tStandard\n",
      "#\t\t'SM_V7_U009'\t42047\tStandard\n",
      "#\t\t'SM_V7_3H006'\t42046\tStandard\n",
      "#\t\t'SM_V7_1H012'\t41993\tStandard\n",
      "#\t\t'SM_V7_1H011'\t41975\tStandard\n",
      "#\t\t'SM_V7_2H012'\t41866\tStandard\n",
      "#\t\t'SM_V7_6H012'\t41854\tStandard\n",
      "#\t\t'SM_V7_U010'\t41849\tStandard\n",
      "#\t\t'SM_V7_1H013'\t41780\tStandard\n",
      "#\t\t'SM_V7_4H011'\t41493\tStandard\n",
      "#\t\t'SM_V7_1H014'\t41416\tStandard\n",
      "#\t\t'SM_V7_1H015'\t40908\tStandard\n",
      "#\t\t'SM_V7_6H011'\t40838\tStandard\n",
      "#\t\t'SM_V7_W017'\t40810\tStandard\n",
      "#\t\t'SM_V7_3H007'\t40564\tStandard\n",
      "#\t\t'SM_V7_U011'\t40472\tStandard\n",
      "#\t\t'SM_V7_1H016'\t40279\tStandard\n",
      "#\t\t'SM_V7_1H017'\t40268\tStandard\n",
      "#\t\t'SM_V7_4H010'\t40207\tStandard\n",
      "#\t\t'SM_V7_1H018'\t40149\tStandard\n",
      "#\t\t'SM_V7_3H008'\t40122\tStandard\n",
      "#\t\t'SM_V7_4H013'\t39978\tStandard\n",
      "#\t\t'SM_V7_1H019'\t39836\tStandard\n",
      "#\t\t'SM_V7_4H012'\t39780\tStandard\n",
      "#\t\t'SM_V7_W021'\t39689\tStandard\n",
      "#\t\t'SM_V7_W020'\t39584\tStandard\n",
      "#\t\t'SM_V7_4H014'\t39472\tStandard\n",
      "#\t\t'SM_V7_ZWH016'\t38948\tStandard\n",
      "#\t\t'SM_V7_4H015'\t38788\tStandard\n",
      "#\t\t'SM_V7_4H016'\t38587\tStandard\n",
      "#\t\t'SM_V7_4H017'\t38450\tStandard\n",
      "#\t\t'SM_V7_ZWH017'\t38440\tStandard\n",
      "#\t\t'SM_V7_3H009'\t38307\tStandard\n",
      "#\t\t'SM_V7_6H014'\t38274\tStandard\n",
      "#\t\t'SM_V7_4H018'\t37883\tStandard\n",
      "#\t\t'SM_V7_3H010'\t37877\tStandard\n",
      "#\t\t'SM_V7_ZWH018'\t37872\tStandard\n",
      "#\t\t'SM_V7_6H013'\t37868\tStandard\n",
      "#\t\t'SM_V7_7H005'\t37848\tStandard\n",
      "#\t\t'SM_V7_4H019'\t37718\tStandard\n",
      "#\t\t'SM_V7_7H006'\t37701\tStandard\n",
      "#\t\t'SM_V7_2H014'\t37647\tStandard\n",
      "#\t\t'SM_V7_5H015'\t37561\tStandard\n",
      "#\t\t'SM_V7_ZWH019'\t37354\tStandard\n",
      "#\t\t'SM_V7_2H013'\t37161\tStandard\n",
      "#\t\t'SM_V7_1H022'\t37152\tStandard\n",
      "#\t\t'SM_V7_1H021'\t37064\tStandard\n",
      "#\t\t'SM_V7_4H021'\t37030\tStandard\n",
      "#\t\t'SM_V7_1H020'\t36937\tStandard\n",
      "#\t\t'SM_V7_4H020'\t36932\tStandard\n",
      "#\t\t'SM_V7_ZWH020'\t36899\tStandard\n",
      "#\t\t'SM_V7_2H015'\t36894\tStandard\n",
      "#\t\t'SM_V7_1H023'\t36773\tStandard\n",
      "#\t\t'SM_V7_5H017'\t36476\tStandard\n",
      "#\t\t'SM_V7_5H016'\t36405\tStandard\n",
      "#\t\t'SM_V7_2H016'\t36330\tStandard\n",
      "#\t\t'SM_V7_3H011'\t36265\tStandard\n",
      "#\t\t'SM_V7_ZWH021'\t36095\tStandard\n",
      "#\t\t'SM_V7_2H018'\t35981\tStandard\n",
      "#\t\t'SM_V7_U013'\t35842\tStandard\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#\t\t'SM_V7_2H017'\t35746\tStandard\n",
      "#\t\t'SM_V7_2H019'\t35649\tStandard\n",
      "#\t\t'SM_V7_1H025'\t35346\tStandard\n",
      "#\t\t'SM_V7_1H024'\t35172\tStandard\n",
      "#\t\t'SM_V7_4H022'\t35161\tStandard\n",
      "#\t\t'SM_V7_ZWU007'\t35161\tStandard\n",
      "#\t\t'SM_V7_4H023'\t34814\tStandard\n",
      "#\t\t'SM_V7_3H012'\t34678\tStandard\n",
      "#\t\t'SM_V7_ZWH023'\t34659\tStandard\n",
      "#\t\t'SM_V7_4H024'\t34614\tStandard\n",
      "#\t\t'SM_V7_1H028'\t34539\tStandard\n",
      "#\t\t'SM_V7_ZWH022'\t34526\tStandard\n",
      "#\t\t'SM_V7_1003'\t34508\tStandard\n",
      "#\t\t'SM_V7_1H026'\t34385\tStandard\n",
      "#\t\t'SM_V7_1H027'\t34385\tStandard\n",
      "#\t\t'SM_V7_7H007'\t34225\tStandard\n",
      "#\t\t'SM_V7_4H025'\t34158\tStandard\n",
      "#\t\t'SM_V7_U012'\t34139\tStandard\n",
      "#\t\t'SM_V7_1H029'\t34050\tStandard\n",
      "#\t\t'SM_V7_5H018'\t33970\tStandard\n",
      "#\t\t'SM_V7_5H019'\t33947\tStandard\n",
      "#\t\t'SM_V7_2H020'\t33720\tStandard\n",
      "#\t\t'SM_V7_2H021'\t33608\tStandard\n",
      "#\t\t'SM_V7_6H015'\t33584\tStandard\n",
      "#\t\t'SM_V7_1H030'\t33580\tStandard\n",
      "#\t\t'SM_V7_4H026'\t33480\tStandard\n",
      "#\t\t'SM_V7_ZWH024'\t33395\tStandard\n",
      "#\t\t'SM_V7_3H014'\t33058\tStandard\n",
      "#\t\t'SM_V7_3H013'\t33002\tStandard\n",
      "#\t\t'SM_V7_5H020'\t32808\tStandard\n",
      "#\t\t'SM_V7_ZWH027'\t32768\tStandard\n",
      "#\t\t'SM_V7_ZWH026'\t32761\tStandard\n",
      "#\t\t'SM_V7_W022'\t32622\tStandard\n",
      "#\t\t'SM_V7_ZWH028'\t32463\tStandard\n",
      "#\t\t'SM_V7_1H032'\t32437\tStandard\n",
      "#\t\t'SM_V7_1H031'\t32322\tStandard\n",
      "#\t\t'SM_V7_3H015'\t32223\tStandard\n",
      "#\t\t'SM_V7_W015'\t32211\tStandard\n",
      "#\t\t'SM_V7_4H027'\t32106\tStandard\n",
      "#\t\t'SM_V7_1H033'\t32089\tStandard\n",
      "#\t\t'SM_V7_1H034'\t32011\tStandard\n",
      "#\t\t'SM_V7_6H016'\t31954\tStandard\n",
      "#\t\t'SM_V7_1H035'\t31914\tStandard\n",
      "#\t\t'SM_V7_ZWH025'\t31897\tStandard\n",
      "#\t\t'SM_V7_4H028'\t31895\tStandard\n",
      "#\t\t'SM_V7_2H022'\t31872\tStandard\n",
      "#\t\t'SM_V7_ZWH029'\t31552\tStandard\n",
      "#\t\t'SM_V7_3H016'\t31458\tStandard\n",
      "#\t\t'SM_V7_4H030'\t31218\tStandard\n",
      "#\t\t'SM_V7_2H023'\t31190\tStandard\n",
      "#\t\t'SM_V7_1H037'\t31153\tStandard\n",
      "#\t\t'SM_V7_2H024'\t31091\tStandard\n",
      "#\t\t'SM_V7_3002'\t30987\tStandard\n",
      "#\t\t'SM_V7_1H036'\t30661\tStandard\n",
      "#\t\t'SM_V7_ZWU006'\t30575\tStandard\n",
      "#\t\t'SM_V7_U014'\t30537\tStandard\n",
      "#\t\t'SM_V7_4H031'\t30457\tStandard\n",
      "#\t\t'SM_V7_4H029'\t30376\tStandard\n",
      "#\t\t'SM_V7_ZWH030'\t30210\tStandard\n",
      "#\t\t'SM_V7_4H032'\t30174\tStandard\n",
      "#\t\t'SM_V7_1H039'\t29978\tStandard\n",
      "#\t\t'SM_V7_2H027'\t29949\tStandard\n",
      "#\t\t'SM_V7_2H026'\t29857\tStandard\n",
      "#\t\t'SM_V7_7H010'\t29794\tStandard\n",
      "#\t\t'SM_V7_3H017'\t29772\tStandard\n",
      "#\t\t'SM_V7_1H041'\t29619\tStandard\n",
      "#\t\t'SM_V7_7H009'\t29420\tStandard\n",
      "#\t\t'SM_V7_1H038'\t29346\tStandard\n",
      "#\t\t'SM_V7_2H025'\t29284\tStandard\n",
      "#\t\t'SM_V7_6H017'\t29216\tStandard\n",
      "#\t\t'SM_V7_ZWH031'\t29206\tStandard\n",
      "#\t\t'SM_V7_7H011'\t28989\tStandard\n",
      "#\t\t'SM_V7_5H021'\t28933\tStandard\n",
      "#\t\t'SM_V7_ZWH032'\t28863\tStandard\n",
      "#\t\t'SM_V7_1H040'\t28404\tStandard\n",
      "#\t\t'SM_V7_ZWH033'\t28372\tStandard\n",
      "#\t\t'SM_V7_5H022'\t28341\tStandard\n",
      "#\t\t'SM_V7_3H018'\t28220\tStandard\n",
      "#\t\t'SM_V7_4H033'\t28174\tStandard\n",
      "#\t\t'SM_V7_3H019'\t28128\tStandard\n",
      "#\t\t'SM_V7_ZWH035'\t27859\tStandard\n",
      "#\t\t'SM_V7_2H028'\t27630\tStandard\n",
      "#\t\t'SM_V7_ZWH036'\t27602\tStandard\n",
      "#\t\t'SM_V7_5H024'\t27421\tStandard\n",
      "#\t\t'SM_V7_5H023'\t27391\tStandard\n",
      "#\t\t'SM_V7_ZWH034'\t27291\tStandard\n",
      "#\t\t'SM_V7_6H018'\t27194\tStandard\n",
      "#\t\t'SM_V7_5H025'\t27052\tStandard\n",
      "#\t\t'SM_V7_1H043'\t27049\tStandard\n",
      "#\t\t'SM_V7_1H042'\t27046\tStandard\n",
      "#\t\t'SM_V7_ZWH037'\t26959\tStandard\n",
      "#\t\t'SM_V7_MITO'\t26917\tTrematode_Mitochondrial\n",
      "#\t\t'SM_V7_4H034'\t26690\tStandard\n",
      "#\t\t'SM_V7_ZWH038'\t26672\tStandard\n",
      "#\t\t'SM_V7_6H019'\t26539\tStandard\n",
      "#\t\t'SM_V7_3H020'\t26491\tStandard\n",
      "#\t\t'SM_V7_6H020'\t26434\tStandard\n",
      "#\t\t'SM_V7_4H035'\t26278\tStandard\n",
      "#\t\t'SM_V7_7H012'\t26102\tStandard\n",
      "#\t\t'SM_V7_1H044'\t25493\tStandard\n",
      "#\t\t'SM_V7_ZWU008'\t25483\tStandard\n",
      "#\t\t'SM_V7_4H036'\t25454\tStandard\n",
      "#\t\t'SM_V7_ZWH039'\t25205\tStandard\n",
      "#\t\t'SM_V7_6H022'\t24884\tStandard\n",
      "#\t\t'SM_V7_6H021'\t24849\tStandard\n",
      "#\t\t'SM_V7_7H013'\t24818\tStandard\n",
      "#\t\t'SM_V7_4H037'\t24441\tStandard\n",
      "#\t\t'SM_V7_7H014'\t24420\tStandard\n",
      "#\t\t'SM_V7_1H045'\t24196\tStandard\n",
      "#\t\t'SM_V7_ZWH041'\t24020\tStandard\n",
      "#\t\t'SM_V7_W023'\t23771\tStandard\n",
      "#\t\t'SM_V7_6H023'\t23718\tStandard\n",
      "#\t\t'SM_V7_3H021'\t23120\tStandard\n",
      "#\t\t'SM_V7_7H008'\t23019\tStandard\n",
      "#\t\t'SM_V7_6H024'\t22726\tStandard\n",
      "#\t\t'SM_V7_ZWH042'\t22423\tStandard\n",
      "#\t\t'SM_V7_6H025'\t22193\tStandard\n",
      "#\t\t'SM_V7_2H030'\t22176\tStandard\n",
      "#\t\t'SM_V7_4H038'\t22085\tStandard\n",
      "#\t\t'SM_V7_ZWH043'\t22068\tStandard\n",
      "#\t\t'SM_V7_7H015'\t21833\tStandard\n",
      "#\t\t'SM_V7_2H029'\t21793\tStandard\n",
      "#\t\t'SM_V7_ZWH044'\t21747\tStandard\n",
      "#\t\t'SM_V7_ZWH045'\t21661\tStandard\n",
      "#\t\t'SM_V7_1H046'\t21617\tStandard\n",
      "#\t\t'SM_V7_2H031'\t21130\tStandard\n",
      "#\t\t'SM_V7_2H032'\t20166\tStandard\n",
      "#\t\t'SM_V7_ZWH040'\t19870\tStandard\n",
      "#\t\t'SM_V7_2H033'\t19679\tStandard\n",
      "#\t\t'SM_V7_U015'\t19106\tStandard\n",
      "#\t\t'SM_V7_2H034'\t18656\tStandard\n",
      "#\t\t'SM_V7_5H026'\t18095\tStandard\n",
      "#\t\t'SM_V7_1H047'\t16498\tStandard\n",
      "#\t\t'SM_V7_U016'\t15741\tStandard\n",
      "#\t\t'SM_V7_2H035'\t14904\tStandard\n",
      "#\t\t'SM_V7_2H036'\t14512\tStandard\n",
      "#\t\t'SM_V7_3H022'\t13856\tStandard\n",
      "#\t\t'SM_V7_6H026'\t13844\tStandard\n",
      "#\t\t'SM_V7_6H027'\t13614\tStandard\n",
      "#\t\t'SM_V7_ZWH046'\t13151\tStandard\n",
      "#\t\t'SM_V7_5H027'\t8381\tStandard\n",
      "#\t\t'SM_V7_U017'\t1307\tStandard\n",
      "#-----------------------------------------------\n",
      "\n",
      "00:00:11\tPredicting variants\n",
      "00:00:11\tCreating sequences from exon information 'SM_V7_1'\n",
      "00:00:12\tBuilding sequence tree for chromosome 'SM_V7_1'\n",
      "00:00:12\tBuilding sequence tree for genome sequences\n",
      "00:00:12\tDone.\n",
      "00:00:12\tDone. Loaded 19808 sequences.\n",
      "00:00:14\t\t10000 variants (2817 variants per second), 9323 VCF entries\n",
      "00:00:16\t\t20000 variants (3678 variants per second), 18684 VCF entries\n",
      "00:00:19\t\t30000 variants (3834 variants per second), 28408 VCF entries\n",
      "00:00:21\t\t40000 variants (4124 variants per second), 37918 VCF entries\n",
      "00:00:23\t\t50000 variants (4238 variants per second), 47683 VCF entries\n",
      "00:00:25\t\t60000 variants (4286 variants per second), 57386 VCF entries\n",
      "00:00:26\t\t70000 variants (4488 variants per second), 67047 VCF entries\n",
      "00:00:28\t\t80000 variants (4542 variants per second), 76617 VCF entries\n",
      "00:00:30\t\t90000 variants (4635 variants per second), 86265 VCF entries\n",
      "00:00:32\t\t100000 variants (4671 variants per second), 95816 VCF entries\n",
      "00:00:34\t\t110000 variants (4697 variants per second), 105464 VCF entries\n",
      "00:00:36\t\t120000 variants (4719 variants per second), 115082 VCF entries\n",
      "00:00:38\t\t130000 variants (4767 variants per second), 124742 VCF entries\n",
      "00:00:40\t\t140000 variants (4741 variants per second), 134257 VCF entries\n",
      "00:00:42\t\t150000 variants (4778 variants per second), 143844 VCF entries\n",
      "00:00:44\t\t160000 variants (4800 variants per second), 153309 VCF entries\n",
      "00:00:46\t\t170000 variants (4795 variants per second), 162829 VCF entries\n",
      "00:00:48\t\t180000 variants (4807 variants per second), 172354 VCF entries\n",
      "00:00:50\t\t190000 variants (4852 variants per second), 181889 VCF entries\n",
      "00:00:52\t\t200000 variants (4860 variants per second), 191494 VCF entries\n",
      "00:00:54\t\t210000 variants (4898 variants per second), 201168 VCF entries\n",
      "00:00:55\t\t220000 variants (4943 variants per second), 210753 VCF entries\n",
      "00:00:57\t\t230000 variants (4969 variants per second), 220221 VCF entries\n",
      "00:00:59\t\t240000 variants (4982 variants per second), 229738 VCF entries\n",
      "00:01:01\t\t250000 variants (5001 variants per second), 239368 VCF entries\n",
      "00:01:03\t\t260000 variants (4987 variants per second), 249013 VCF entries\n",
      "00:01:05\t\t270000 variants (4993 variants per second), 258594 VCF entries\n",
      "00:01:07\t\t280000 variants (4994 variants per second), 268269 VCF entries\n",
      "00:01:09\t\t290000 variants (5004 variants per second), 277844 VCF entries\n",
      "00:01:11\t\t300000 variants (5007 variants per second), 287406 VCF entries\n",
      "00:01:13\t\t310000 variants (5011 variants per second), 297122 VCF entries\n",
      "00:01:14\t\t320000 variants (5031 variants per second), 306814 VCF entries\n",
      "00:01:16\t\t330000 variants (5047 variants per second), 316403 VCF entries\n",
      "00:01:18\t\t340000 variants (5052 variants per second), 326177 VCF entries\n",
      "00:01:20\t\t350000 variants (5073 variants per second), 335847 VCF entries\n",
      "00:01:21\tCreating sequences from exon information 'SM_V7_ZW'\n",
      "00:01:21\tBuilding sequence tree for chromosome 'SM_V7_ZW'\n",
      "00:01:21\tBuilding sequence tree for genome sequences\n",
      "00:01:21\tDone.\n",
      "00:01:21\tDone. Loaded 19363 sequences.\n",
      "00:01:23\t\t360000 variants (5010 variants per second), 345580 VCF entries\n",
      "00:01:25\t\t370000 variants (4987 variants per second), 355322 VCF entries\n",
      "00:01:27\t\t380000 variants (5000 variants per second), 365093 VCF entries\n",
      "00:01:29\t\t390000 variants (5017 variants per second), 374879 VCF entries\n",
      "00:01:31\t\t400000 variants (5021 variants per second), 384641 VCF entries\n",
      "00:01:33\t\t410000 variants (5011 variants per second), 394193 VCF entries\n",
      "00:01:35\t\t420000 variants (5002 variants per second), 403582 VCF entries\n",
      "00:01:37\t\t430000 variants (4999 variants per second), 412915 VCF entries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:01:39\t\t440000 variants (4995 variants per second), 422530 VCF entries\n",
      "00:01:41\t\t450000 variants (4986 variants per second), 431965 VCF entries\n",
      "00:01:43\t\t460000 variants (4988 variants per second), 441404 VCF entries\n",
      "00:01:45\t\t470000 variants (4990 variants per second), 450925 VCF entries\n",
      "00:01:47\t\t480000 variants (4985 variants per second), 460470 VCF entries\n",
      "00:01:49\t\t490000 variants (5002 variants per second), 469964 VCF entries\n",
      "00:01:51\t\t500000 variants (4998 variants per second), 479496 VCF entries\n",
      "00:01:53\t\t510000 variants (5005 variants per second), 489149 VCF entries\n",
      "00:01:55\t\t520000 variants (5004 variants per second), 498766 VCF entries\n",
      "00:01:57\t\t530000 variants (5015 variants per second), 508390 VCF entries\n",
      "00:01:58\t\t540000 variants (5023 variants per second), 518087 VCF entries\n",
      "00:02:00\t\t550000 variants (5022 variants per second), 527594 VCF entries\n",
      "00:02:02\t\t560000 variants (5021 variants per second), 537323 VCF entries\n",
      "00:02:04\t\t570000 variants (5030 variants per second), 546976 VCF entries\n",
      "00:02:06\t\t580000 variants (5025 variants per second), 556590 VCF entries\n",
      "00:02:08\t\t590000 variants (5029 variants per second), 566160 VCF entries\n",
      "00:02:10\t\t600000 variants (5040 variants per second), 575730 VCF entries\n",
      "00:02:12\t\t610000 variants (5041 variants per second), 585356 VCF entries\n",
      "00:02:14\t\t620000 variants (5049 variants per second), 595038 VCF entries\n",
      "00:02:16\t\t630000 variants (5049 variants per second), 604788 VCF entries\n",
      "00:02:18\t\t640000 variants (5045 variants per second), 614522 VCF entries\n",
      "00:02:19\t\t650000 variants (5061 variants per second), 624193 VCF entries\n",
      "00:02:20\tCreating sequences from exon information 'SM_V7_3'\n",
      "00:02:20\tBuilding sequence tree for chromosome 'SM_V7_3'\n",
      "00:02:20\tBuilding sequence tree for genome sequences\n",
      "00:02:20\tDone.\n",
      "00:02:20\tDone. Loaded 10153 sequences.\n",
      "00:02:21\t\t660000 variants (5056 variants per second), 633945 VCF entries\n",
      "00:02:23\t\t670000 variants (5063 variants per second), 643739 VCF entries\n",
      "00:02:25\t\t680000 variants (5073 variants per second), 653605 VCF entries\n",
      "00:02:27\t\t690000 variants (5081 variants per second), 663403 VCF entries\n",
      "00:02:29\t\t700000 variants (5083 variants per second), 673238 VCF entries\n",
      "00:02:30\t\t710000 variants (5086 variants per second), 683023 VCF entries\n",
      "00:02:32\t\t720000 variants (5085 variants per second), 692653 VCF entries\n",
      "00:02:35\t\t730000 variants (5080 variants per second), 702191 VCF entries\n",
      "00:02:36\t\t740000 variants (5088 variants per second), 711842 VCF entries\n",
      "00:02:38\t\t750000 variants (5086 variants per second), 721493 VCF entries\n",
      "00:02:40\t\t760000 variants (5089 variants per second), 731022 VCF entries\n",
      "00:02:42\t\t770000 variants (5095 variants per second), 740688 VCF entries\n",
      "00:02:44\t\t780000 variants (5095 variants per second), 750085 VCF entries\n",
      "00:02:46\t\t790000 variants (5082 variants per second), 759625 VCF entries\n",
      "00:02:48\t\t800000 variants (5088 variants per second), 769048 VCF entries\n",
      "00:02:50\t\t810000 variants (5092 variants per second), 778694 VCF entries\n",
      "00:02:52\t\t820000 variants (5099 variants per second), 788395 VCF entries\n",
      "00:02:54\t\t830000 variants (5102 variants per second), 798028 VCF entries\n",
      "00:02:55\t\t840000 variants (5108 variants per second), 807714 VCF entries\n",
      "00:02:57\t\t850000 variants (5100 variants per second), 817424 VCF entries\n",
      "00:03:00\t\t860000 variants (5098 variants per second), 827167 VCF entries\n",
      "00:03:01\t\t870000 variants (5103 variants per second), 836802 VCF entries\n",
      "00:03:03\t\t880000 variants (5106 variants per second), 846474 VCF entries\n",
      "00:03:05\t\t890000 variants (5114 variants per second), 855998 VCF entries\n",
      "00:03:07\t\t900000 variants (5115 variants per second), 865639 VCF entries\n",
      "00:03:09\tCreating sequences from exon information 'SM_V7_2'\n",
      "00:03:09\tBuilding sequence tree for chromosome 'SM_V7_2'\n",
      "00:03:09\tBuilding sequence tree for genome sequences\n",
      "00:03:09\tDone.\n",
      "00:03:09\tDone. Loaded 10230 sequences.\n",
      "00:03:09\t\t910000 variants (5107 variants per second), 875042 VCF entries\n",
      "00:03:11\t\t920000 variants (5107 variants per second), 884849 VCF entries\n",
      "00:03:13\t\t930000 variants (5115 variants per second), 894611 VCF entries\n",
      "00:03:15\t\t940000 variants (5114 variants per second), 904334 VCF entries\n",
      "00:03:16\t\t950000 variants (5119 variants per second), 914070 VCF entries\n",
      "00:03:18\t\t960000 variants (5130 variants per second), 923815 VCF entries\n",
      "00:03:20\t\t970000 variants (5140 variants per second), 933593 VCF entries\n",
      "00:03:21\t\t980000 variants (5147 variants per second), 943415 VCF entries\n",
      "00:03:23\t\t990000 variants (5156 variants per second), 953193 VCF entries\n",
      "00:03:25\t\t1000000 variants (5160 variants per second), 962899 VCF entries\n",
      "00:03:26\t\t1010000 variants (5172 variants per second), 972674 VCF entries\n",
      "00:03:28\t\t1020000 variants (5176 variants per second), 982219 VCF entries\n",
      "00:03:30\t\t1030000 variants (5179 variants per second), 991974 VCF entries\n",
      "00:03:31\t\t1040000 variants (5185 variants per second), 1001745 VCF entries\n",
      "00:03:33\t\t1050000 variants (5191 variants per second), 1011376 VCF entries\n",
      "00:03:35\t\t1060000 variants (5193 variants per second), 1020993 VCF entries\n",
      "00:03:37\t\t1070000 variants (5197 variants per second), 1030654 VCF entries\n",
      "00:03:38\t\t1080000 variants (5203 variants per second), 1040284 VCF entries\n",
      "00:03:41\t\t1090000 variants (5196 variants per second), 1049866 VCF entries\n",
      "00:03:43\t\t1100000 variants (5189 variants per second), 1059474 VCF entries\n",
      "00:03:45\t\t1110000 variants (5187 variants per second), 1069251 VCF entries\n",
      "00:03:47\t\t1120000 variants (5187 variants per second), 1078995 VCF entries\n",
      "00:03:48\t\t1130000 variants (5192 variants per second), 1088768 VCF entries\n",
      "00:03:50\t\t1140000 variants (5196 variants per second), 1098413 VCF entries\n",
      "00:03:52\t\t1150000 variants (5199 variants per second), 1108126 VCF entries\n",
      "00:03:54\t\t1160000 variants (5200 variants per second), 1117931 VCF entries\n",
      "00:03:55\tCreating sequences from exon information 'SM_V7_4'\n",
      "00:03:55\tBuilding sequence tree for chromosome 'SM_V7_4'\n",
      "00:03:55\tBuilding sequence tree for genome sequences\n",
      "00:03:55\tDone.\n",
      "00:03:55\tDone. Loaded 9616 sequences.\n",
      "00:03:56\t\t1170000 variants (5204 variants per second), 1127631 VCF entries\n",
      "00:03:57\t\t1180000 variants (5209 variants per second), 1137442 VCF entries\n",
      "00:03:59\t\t1190000 variants (5213 variants per second), 1147330 VCF entries\n",
      "00:04:01\t\t1200000 variants (5204 variants per second), 1157167 VCF entries\n",
      "00:04:03\t\t1210000 variants (5210 variants per second), 1166635 VCF entries\n",
      "00:04:05\t\t1220000 variants (5209 variants per second), 1176093 VCF entries\n",
      "00:04:07\t\t1230000 variants (5207 variants per second), 1185700 VCF entries\n",
      "00:04:09\t\t1240000 variants (5211 variants per second), 1195258 VCF entries\n",
      "00:04:11\t\t1250000 variants (5214 variants per second), 1204811 VCF entries\n",
      "00:04:12\t\t1260000 variants (5219 variants per second), 1214471 VCF entries\n",
      "00:04:14\t\t1270000 variants (5224 variants per second), 1223896 VCF entries\n",
      "00:04:16\t\t1280000 variants (5224 variants per second), 1233420 VCF entries\n",
      "00:04:18\t\t1290000 variants (5227 variants per second), 1243068 VCF entries\n",
      "00:04:20\t\t1300000 variants (5228 variants per second), 1252443 VCF entries\n",
      "00:04:21\t\t1310000 variants (5230 variants per second), 1262137 VCF entries\n",
      "00:04:23\t\t1320000 variants (5227 variants per second), 1271778 VCF entries\n",
      "00:04:25\t\t1330000 variants (5228 variants per second), 1281545 VCF entries\n",
      "00:04:27\t\t1340000 variants (5232 variants per second), 1291475 VCF entries\n",
      "00:04:29\t\t1350000 variants (5236 variants per second), 1301206 VCF entries\n",
      "00:04:30\t\t1360000 variants (5241 variants per second), 1310951 VCF entries\n",
      "00:04:32\t\t1370000 variants (5245 variants per second), 1320679 VCF entries\n",
      "00:04:32\tCreating sequences from exon information 'SM_V7_5'\n",
      "00:04:32\tBuilding sequence tree for chromosome 'SM_V7_5'\n",
      "00:04:32\tBuilding sequence tree for genome sequences\n",
      "00:04:32\tDone.\n",
      "00:04:32\tDone. Loaded 4632 sequences.\n",
      "00:04:34\t\t1380000 variants (5251 variants per second), 1330514 VCF entries\n",
      "00:04:35\t\t1390000 variants (5263 variants per second), 1340424 VCF entries\n",
      "00:04:36\t\t1400000 variants (5270 variants per second), 1350305 VCF entries\n",
      "00:04:38\t\t1410000 variants (5275 variants per second), 1360062 VCF entries\n",
      "00:04:40\t\t1420000 variants (5281 variants per second), 1369745 VCF entries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:04:41\t\t1430000 variants (5283 variants per second), 1379558 VCF entries\n",
      "00:04:43\t\t1440000 variants (5286 variants per second), 1389333 VCF entries\n",
      "00:04:45\t\t1450000 variants (5291 variants per second), 1398821 VCF entries\n",
      "00:04:47\t\t1460000 variants (5294 variants per second), 1408477 VCF entries\n",
      "00:04:48\t\t1470000 variants (5295 variants per second), 1418221 VCF entries\n",
      "00:04:50\t\t1480000 variants (5295 variants per second), 1427989 VCF entries\n",
      "00:04:52\t\t1490000 variants (5296 variants per second), 1437751 VCF entries\n",
      "00:04:54\t\t1500000 variants (5296 variants per second), 1447564 VCF entries\n",
      "00:04:56\t\t1510000 variants (5299 variants per second), 1457272 VCF entries\n",
      "00:04:57\t\t1520000 variants (5302 variants per second), 1467103 VCF entries\n",
      "00:04:59\t\t1530000 variants (5304 variants per second), 1476726 VCF entries\n",
      "00:05:01\t\t1540000 variants (5309 variants per second), 1486404 VCF entries\n",
      "00:05:01\tCreating sequences from exon information 'SM_V7_6'\n",
      "00:05:01\tBuilding sequence tree for chromosome 'SM_V7_6'\n",
      "00:05:01\tBuilding sequence tree for genome sequences\n",
      "00:05:01\tDone.\n",
      "00:05:01\tDone. Loaded 5156 sequences.\n",
      "00:05:03\t\t1550000 variants (5309 variants per second), 1496093 VCF entries\n",
      "00:05:05\t\t1560000 variants (5304 variants per second), 1505394 VCF entries\n",
      "00:05:07\t\t1570000 variants (5301 variants per second), 1515138 VCF entries\n",
      "00:05:09\t\t1580000 variants (5301 variants per second), 1524828 VCF entries\n",
      "00:05:11\t\t1590000 variants (5301 variants per second), 1534553 VCF entries\n",
      "00:05:13\t\t1600000 variants (5299 variants per second), 1544186 VCF entries\n",
      "00:05:15\t\t1610000 variants (5299 variants per second), 1553744 VCF entries\n",
      "00:05:16\t\t1620000 variants (5304 variants per second), 1563451 VCF entries\n",
      "00:05:18\t\t1630000 variants (5309 variants per second), 1573208 VCF entries\n",
      "00:05:20\t\t1640000 variants (5308 variants per second), 1582863 VCF entries\n",
      "00:05:22\t\t1650000 variants (5304 variants per second), 1592570 VCF entries\n",
      "00:05:24\t\t1660000 variants (5303 variants per second), 1602354 VCF entries\n",
      "00:05:26\t\t1670000 variants (5306 variants per second), 1612264 VCF entries\n",
      "00:05:28\t\t1680000 variants (5304 variants per second), 1622137 VCF entries\n",
      "00:05:29\t\t1690000 variants (5303 variants per second), 1631814 VCF entries\n",
      "00:05:32\t\t1700000 variants (5298 variants per second), 1641553 VCF entries\n",
      "00:05:33\tCreating sequences from exon information 'SM_V7_7'\n",
      "00:05:34\tBuilding sequence tree for chromosome 'SM_V7_7'\n",
      "00:05:34\tBuilding sequence tree for genome sequences\n",
      "00:05:34\tDone.\n",
      "00:05:34\tDone. Loaded 3904 sequences.\n",
      "00:05:34\t\t1710000 variants (5296 variants per second), 1651322 VCF entries\n",
      "00:05:35\t\t1720000 variants (5300 variants per second), 1661180 VCF entries\n",
      "00:05:37\t\t1730000 variants (5302 variants per second), 1670881 VCF entries\n",
      "00:05:39\t\t1740000 variants (5304 variants per second), 1680633 VCF entries\n",
      "00:05:41\t\t1750000 variants (5306 variants per second), 1690374 VCF entries\n",
      "00:05:42\t\t1760000 variants (5308 variants per second), 1700084 VCF entries\n",
      "00:05:44\t\t1770000 variants (5310 variants per second), 1709866 VCF entries\n",
      "00:05:46\t\t1780000 variants (5314 variants per second), 1719716 VCF entries\n",
      "00:05:47\t\t1790000 variants (5317 variants per second), 1729433 VCF entries\n",
      "00:05:49\t\t1800000 variants (5319 variants per second), 1739195 VCF entries\n",
      "00:05:51\t\t1810000 variants (5318 variants per second), 1748971 VCF entries\n",
      "00:05:52\tCreating sequences from exon information 'SM_V7_W001'\n",
      "00:05:52\tBuilding sequence tree for chromosome 'SM_V7_W001'\n",
      "00:05:52\tBuilding sequence tree for genome sequences\n",
      "00:05:52\tDone.\n",
      "00:05:52\tDone. Loaded 55 sequences.\n",
      "00:05:52\tCreating sequences from exon information 'SM_V7_W002'\n",
      "00:05:52\tBuilding sequence tree for chromosome 'SM_V7_W002'\n",
      "00:05:52\tBuilding sequence tree for genome sequences\n",
      "00:05:52\tDone.\n",
      "00:05:52\tDone. Loaded 14 sequences.\n",
      "00:05:52\tCreating sequences from exon information 'SM_V7_W003'\n",
      "00:05:52\tBuilding sequence tree for chromosome 'SM_V7_W003'\n",
      "00:05:52\tBuilding sequence tree for genome sequences\n",
      "00:05:52\tDone.\n",
      "00:05:52\tDone. Loaded 17 sequences.\n",
      "00:05:52\tCreating sequences from exon information 'SM_V7_ZWU001'\n",
      "00:05:52\tBuilding sequence tree for chromosome 'SM_V7_ZWU001'\n",
      "00:05:52\tBuilding sequence tree for genome sequences\n",
      "00:05:52\tDone.\n",
      "00:05:52\tDone. Loaded 24 sequences.\n",
      "00:05:53\tCreating sequences from exon information 'SM_V7_W004'\n",
      "00:05:53\tBuilding sequence tree for chromosome 'SM_V7_W004'\n",
      "00:05:53\tBuilding sequence tree for genome sequences\n",
      "00:05:53\tDone.\n",
      "00:05:53\tDone. Loaded 9 sequences.\n",
      "00:05:53\tCreating sequences from exon information 'SM_V7_5H001'\n",
      "00:05:53\tBuilding sequence tree for chromosome 'SM_V7_5H001'\n",
      "00:05:53\tBuilding sequence tree for genome sequences\n",
      "00:05:53\tDone.\n",
      "00:05:53\tDone. Loaded 30 sequences.\n",
      "00:05:53\tCreating sequences from exon information 'SM_V7_2H001'\n",
      "00:05:53\tBuilding sequence tree for chromosome 'SM_V7_2H001'\n",
      "00:05:53\tBuilding sequence tree for genome sequences\n",
      "00:05:53\tDone.\n",
      "00:05:53\tDone. Loaded 32 sequences.\n",
      "00:05:53\t\t1820000 variants (5319 variants per second), 1758654 VCF entries\n",
      "00:05:54\tCreating sequences from exon information 'SM_V7_ZWU002'\n",
      "00:05:54\tBuilding sequence tree for chromosome 'SM_V7_ZWU002'\n",
      "00:05:54\tBuilding sequence tree for genome sequences\n",
      "00:05:54\tDone.\n",
      "00:05:54\tDone. Loaded 42 sequences.\n",
      "00:05:54\tCreating sequences from exon information 'SM_V7_2H002'\n",
      "00:05:54\tBuilding sequence tree for chromosome 'SM_V7_2H002'\n",
      "00:05:54\tBuilding sequence tree for genome sequences\n",
      "00:05:54\tDone.\n",
      "00:05:54\tDone. Loaded 53 sequences.\n",
      "00:05:54\tCreating sequences from exon information 'SM_V7_ZWH001'\n",
      "00:05:54\tBuilding sequence tree for chromosome 'SM_V7_ZWH001'\n",
      "00:05:54\tBuilding sequence tree for genome sequences\n",
      "00:05:54\tDone.\n",
      "00:05:54\tDone. Loaded 55 sequences.\n",
      "00:05:54\tCreating sequences from exon information 'SM_V7_ZWU003'\n",
      "00:05:54\tBuilding sequence tree for chromosome 'SM_V7_ZWU003'\n",
      "00:05:54\tBuilding sequence tree for genome sequences\n",
      "00:05:54\tDone.\n",
      "00:05:54\tDone. Loaded 1 sequences.\n",
      "00:05:54\tCreating sequences from exon information 'SM_V7_W006'\n",
      "00:05:54\tBuilding sequence tree for chromosome 'SM_V7_W006'\n",
      "00:05:54\tBuilding sequence tree for genome sequences\n",
      "00:05:54\tDone.\n",
      "00:05:54\tDone. Loaded 28 sequences.\n",
      "00:05:54\tCreating sequences from exon information 'SM_V7_2H003'\n",
      "00:05:54\tBuilding sequence tree for chromosome 'SM_V7_2H003'\n",
      "00:05:54\tBuilding sequence tree for genome sequences\n",
      "00:05:54\tDone.\n",
      "00:05:54\tDone. Loaded 15 sequences.\n",
      "00:05:55\t\t1830000 variants (5323 variants per second), 1768205 VCF entries\n",
      "00:05:55\tCreating sequences from exon information 'SM_V7_2H004'\n",
      "00:05:55\tBuilding sequence tree for chromosome 'SM_V7_2H004'\n",
      "00:05:55\tBuilding sequence tree for genome sequences\n",
      "00:05:55\tDone.\n",
      "00:05:55\tDone. Loaded 18 sequences.\n",
      "00:05:55\tCreating sequences from exon information 'SM_V7_2H005'\n",
      "00:05:55\tBuilding sequence tree for chromosome 'SM_V7_2H005'\n",
      "00:05:55\tBuilding sequence tree for genome sequences\n",
      "00:05:55\tDone.\n",
      "00:05:55\tDone. Loaded 16 sequences.\n",
      "00:05:55\tCreating sequences from exon information 'SM_V7_U002'\n",
      "00:05:55\tBuilding sequence tree for chromosome 'SM_V7_U002'\n",
      "00:05:55\tBuilding sequence tree for genome sequences\n",
      "00:05:55\tDone.\n",
      "00:05:55\tDone. Loaded 12 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_2H006'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_2H006'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 20 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_4H001'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_4H001'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 12 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_7H001'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_7H001'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 13 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_4H002'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_4H002'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 30 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_W009'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_W009'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 20 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_U003'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_U003'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 27 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_2H007'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_2H007'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 49 sequences.\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_6H001'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_6H001'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 8 sequences.\n",
      "00:05:56\t\t1840000 variants (5326 variants per second), 1777944 VCF entries\n",
      "00:05:56\tCreating sequences from exon information 'SM_V7_5H002'\n",
      "00:05:56\tBuilding sequence tree for chromosome 'SM_V7_5H002'\n",
      "00:05:56\tBuilding sequence tree for genome sequences\n",
      "00:05:56\tDone.\n",
      "00:05:56\tDone. Loaded 23 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_7H002'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_7H002'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 39 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_3H001'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_3H001'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 25 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_U004'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_U004'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 10 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_2H008'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_2H008'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 15 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_W011'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_W011'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 5 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_6H002'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_6H002'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 11 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_2H009'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_2H009'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 18 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_5H005'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_5H005'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 24 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_5H008'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_5H008'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 19 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_5H007'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_5H007'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 7 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_ZWH003'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_ZWH003'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 14 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_6H003'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_6H003'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 10 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_1H002'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_1H002'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 19 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_1H001'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_1H001'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 15 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_6H004'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_6H004'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 24 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_5H009'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_5H009'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 4 sequences.\n",
      "00:05:57\tCreating sequences from exon information 'SM_V7_ZWH004'\n",
      "00:05:57\tBuilding sequence tree for chromosome 'SM_V7_ZWH004'\n",
      "00:05:57\tBuilding sequence tree for genome sequences\n",
      "00:05:57\tDone.\n",
      "00:05:57\tDone. Loaded 4 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_4H003'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_4H003'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 5 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_ZWH007'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_ZWH007'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 30 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_ZWH005'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_ZWH005'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 15 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_7H003'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_7H003'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 2 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_5H010'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_5H010'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 7 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_ZWH009'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_ZWH009'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 18 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_1H003'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_1H003'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 15 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_ZWU004'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_ZWU004'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 11 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_5H011'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_5H011'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 13 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_U005'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_U005'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 3 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_6H006'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_6H006'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 15 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_2H010'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_2H010'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 20 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_6H007'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_6H007'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 12 sequences.\n",
      "00:05:58\t\t1850000 variants (5326 variants per second), 1787754 VCF entries\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_7H004'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_7H004'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 13 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_4H004'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_4H004'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 15 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_1H004'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_1H004'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 11 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_6H008'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_6H008'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 14 sequences.\n",
      "00:05:58\tCreating sequences from exon information 'SM_V7_5H012'\n",
      "00:05:58\tBuilding sequence tree for chromosome 'SM_V7_5H012'\n",
      "00:05:58\tBuilding sequence tree for genome sequences\n",
      "00:05:58\tDone.\n",
      "00:05:58\tDone. Loaded 17 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_2H011'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_2H011'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 24 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_U007'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_U007'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 8 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_W016'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_W016'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 2 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_1H007'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_1H007'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 19 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_W019'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_W019'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 18 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_1H006'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_1H006'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 12 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_ZWH010'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_ZWH010'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 2 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_4H006'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_4H006'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 8 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_5H013'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_5H013'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 12 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_3H003'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_3H003'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 11 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_ZWU005'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_ZWU005'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 7 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_4H007'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_4H007'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 16 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_U008'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_U008'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 9 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_ZWH012'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_ZWH012'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 6 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_6H009'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_6H009'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 10 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_1H009'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_1H009'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 8 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_1H010'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_1H010'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 8 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_3H004'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_3H004'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 14 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_6H010'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_6H010'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 5 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_ZWH014'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_ZWH014'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 17 sequences.\n",
      "00:05:59\tCreating sequences from exon information 'SM_V7_ZWH015'\n",
      "00:05:59\tBuilding sequence tree for chromosome 'SM_V7_ZWH015'\n",
      "00:05:59\tBuilding sequence tree for genome sequences\n",
      "00:05:59\tDone.\n",
      "00:05:59\tDone. Loaded 15 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1002'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1002'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 5 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H009'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H009'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 11 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_3H006'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_3H006'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 13 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H011'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H011'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 11 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_2H012'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_2H012'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 6 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_6H012'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_6H012'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 10 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_U010'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_U010'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 8 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H014'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H014'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 6 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H015'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H015'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 2 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H016'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H016'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 20 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H017'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H017'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 4 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H018'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H018'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 2 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_3H008'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_3H008'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 6 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H019'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H019'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 18 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H012'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H012'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 18 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_W020'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_W020'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 6 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H014'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H014'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 9 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H016'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H016'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 10 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H017'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H017'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 2 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_ZWH017'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_ZWH017'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 5 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_3H009'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_3H009'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 10 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H018'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H018'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 8 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_ZWH018'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_ZWH018'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 9 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_6H013'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_6H013'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 10 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_7H005'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_7H005'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 3 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H019'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H019'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 1 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_7H006'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_7H006'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 5 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_2H014'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_2H014'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 5 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_5H015'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_5H015'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 3 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_ZWH019'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_ZWH019'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 12 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_5H017'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_5H017'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 6 sequences.\n",
      "00:06:00\t\t1860000 variants (5325 variants per second), 1797496 VCF entries\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_3H011'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_3H011'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 16 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_U013'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_U013'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 6 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_2H019'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_2H019'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 10 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H024'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H024'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 17 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_ZWH023'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_ZWH023'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 15 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1003'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1003'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 6 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H027'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H027'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 10 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_7H007'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_7H007'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 12 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_4H025'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_4H025'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 13 sequences.\n",
      "00:06:00\tCreating sequences from exon information 'SM_V7_1H029'\n",
      "00:06:00\tBuilding sequence tree for chromosome 'SM_V7_1H029'\n",
      "00:06:00\tBuilding sequence tree for genome sequences\n",
      "00:06:00\tDone.\n",
      "00:06:00\tDone. Loaded 9 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_2H021'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_2H021'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 6 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_6H015'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_6H015'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 3 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_1H030'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_1H030'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 2 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H026'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H026'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_3H013'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_3H013'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_3H015'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_3H015'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 14 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H027'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H027'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH025'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH025'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 7 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH029'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH029'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 4 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_3H016'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_3H016'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H030'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H030'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 4 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_2H024'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_2H024'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 19 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_1H036'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_1H036'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 4 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWU006'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWU006'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H031'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H031'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 1 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH030'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH030'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 4 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H032'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H032'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 2 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_2H026'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_2H026'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 9 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_3H017'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_3H017'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 3 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_2H025'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_2H025'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 4 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_6H017'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_6H017'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 9 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH033'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH033'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 6 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_5H022'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_5H022'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 7 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H033'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H033'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 9 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_6H018'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_6H018'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 2 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_1H043'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_1H043'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_MITO'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_MITO'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 20 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_6H019'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_6H019'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 9 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_6H020'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_6H020'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H035'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H035'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 16 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_7H012'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_7H012'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 11 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_6H022'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_6H022'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 6 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_4H037'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_4H037'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 6 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_7H014'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_7H014'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 4 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_1H045'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_1H045'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 6 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH041'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH041'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 5 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH042'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH042'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 2 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_2H030'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_2H030'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 3 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH043'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH043'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 7 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_ZWH044'\n",
      "00:06:01\tBuilding sequence tree for chromosome 'SM_V7_ZWH044'\n",
      "00:06:01\tBuilding sequence tree for genome sequences\n",
      "00:06:01\tDone.\n",
      "00:06:01\tDone. Loaded 4 sequences.\n",
      "00:06:01\tCreating sequences from exon information 'SM_V7_2H031'\n",
      "00:06:02\tBuilding sequence tree for chromosome 'SM_V7_2H031'\n",
      "00:06:02\tBuilding sequence tree for genome sequences\n",
      "00:06:02\tDone.\n",
      "00:06:02\tDone. Loaded 7 sequences.\n",
      "00:06:02\tCreating sequences from exon information 'SM_V7_ZWH040'\n",
      "00:06:02\tBuilding sequence tree for chromosome 'SM_V7_ZWH040'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:06:02\tBuilding sequence tree for genome sequences\n",
      "00:06:02\tDone.\n",
      "00:06:02\tDone. Loaded 5 sequences.\n",
      "00:06:02\tCreating sequences from exon information 'SM_V7_U015'\n",
      "00:06:02\tBuilding sequence tree for chromosome 'SM_V7_U015'\n",
      "00:06:02\tBuilding sequence tree for genome sequences\n",
      "00:06:02\tDone.\n",
      "00:06:02\tDone. Loaded 4 sequences.\n",
      "00:06:02\tCreating sequences from exon information 'SM_V7_2H036'\n",
      "00:06:02\tBuilding sequence tree for chromosome 'SM_V7_2H036'\n",
      "00:06:02\tBuilding sequence tree for genome sequences\n",
      "00:06:02\tDone.\n",
      "00:06:02\tDone. Loaded 3 sequences.\n",
      "00:06:02\tCreating sequences from exon information 'SM_V7_6H027'\n",
      "00:06:02\tBuilding sequence tree for chromosome 'SM_V7_6H027'\n",
      "00:06:02\tBuilding sequence tree for genome sequences\n",
      "00:06:02\tDone.\n",
      "00:06:02\tDone. Loaded 3 sequences.\n",
      "00:06:02\tCreating sequences from exon information 'SM_V7_ZWH046'\n",
      "00:06:02\tBuilding sequence tree for chromosome 'SM_V7_ZWH046'\n",
      "00:06:02\tBuilding sequence tree for genome sequences\n",
      "00:06:02\tDone.\n",
      "00:06:02\tDone. Loaded 1 sequences.\n",
      "\n",
      "WARNINGS: Some warning were detected\n",
      "Warning type\tNumber of warnings\n",
      "INFO_REALIGN_3_PRIME\t9690\n",
      "WARNING_TRANSCRIPT_INCOMPLETE\t7\n",
      "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t97\n",
      "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS&INFO_REALIGN_3_PRIME\t1\n",
      "WARNING_TRANSCRIPT_NO_START_CODON\t6922\n",
      "WARNING_TRANSCRIPT_NO_START_CODON&INFO_REALIGN_3_PRIME\t18\n",
      "WARNING_TRANSCRIPT_NO_STOP_CODON\t81\n",
      "WARNING_TRANSCRIPT_NO_STOP_CODON&INFO_REALIGN_3_PRIME\t1\n",
      "\n",
      "\n",
      "00:06:02\tCreating summary file: snpEff_summary.html\n",
      "00:06:03\tCreating genes file: snpEff_genes.txt\n",
      "00:06:05\tdone.\n",
      "00:06:05\tLogging\n",
      "00:06:06\tChecking for updates...\n",
      "00:06:07\tDone.\n",
      "(PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Status directory\n",
    "cdir=\"data/calling\"\n",
    "[[ ! -d \"$cdir\" ]] && mkdir -p \"$cdir\"\n",
    "\n",
    "myvcf=$(grep -m 1 VCF_PFX GWAS.smk | cut -d \"\\\"\" -f 2)\n",
    "\n",
    "snpEff -v -c \"$sedir/snpEff.config\" smansoni \"$cdir/$myvcf.vcf.gz\" | bgzip -c -@ $(nproc) > \"$cdir/${myvcf}_snpEff.vcf.gz\"\n",
    "\n",
    "# Index the VCF file\n",
    "tabix \"$cdir/${myvcf}_snpEff.vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-28T00:38:27.690528Z",
     "start_time": "2021-01-28T00:38:27.133354Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Move file in the right folder\n",
    "mv -t \"$cdir/\" snpEff_summary.* snpEff_genes.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## QTL analysis\n",
    "\n",
    "The analysis relies on several steps:\n",
    "* **SNP filtration**: bi-allelic site only, read depth (DP) of 10 at least, genotype quality (GQ) of 30.\n",
    "* **Conversion of the genotype data in R/qtl format (csvs)**: only alternatively fixed alleles from the parent retained.\n",
    "* **Single QTL analysis on all the phenotypes** (estimated run time: 30 minutes): sum of cercariae shed by schistosome genotype, average of cercariae shed by schistosome genotype, total phenoloxidase activity of the snail host, hemoglobin rate of the snail host.\n",
    "* **QTL interaction analysis using reduced set of markers** (estimated run time: 12h).\n",
    "* **Allele dominance analysis.**\n",
    "\n",
    "N.B.: Estimated run time indicated is dependent on computer/server specifications. Our analysis ran on a server with 16 CPU (2.53 GHz) and 96 GB of RAM."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Rscript scripts/Cercariae_intensity_qtl.R\n",
    "Rscript scripts/Fig4_plot_dominance_chr.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-15T22:24:10.937291Z",
     "start_time": "2021-04-15T22:24:10.827472Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a\n",
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "echo a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prioritization of candidate genes\n",
    "\n",
    "### Chromosome level VCF\n",
    "\n",
    "QTLs were identified on chromosomes 1, 2, 3, 4 and 5. The VCF file needs to be reduced to chromosome intervals to avoid loading unnecessary data and saving time."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-04T16:31:43.773779Z",
     "start_time": "2020-08-04T16:22:41.772466Z"
    }
   },
   "outputs": [],
   "source": [
    "gatk SelectVariants \\\n",
    "        -V \"data/calling/cerc_prod_snpEff.vcf.gz\" \\\n",
    "        -L SM_V7_1:40000000-55000000 \\\n",
    "        -L SM_V7_2:40000000-48130368 \\ # End of the chromosome\n",
    "        -L SM_V7_3:3000000-5500000 \\\n",
    "        -L SM_V7_4:20000000-40000000 \\\n",
    "        -L SM_V7_5:18000000-25000000 \\\n",
    "        -O \"data/calling/cerc_prod_snpEff_1-2-3-4-5.vcf.gz\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Candidate gene analysis\n",
    "\n",
    "The 5 QTLs identified cover several Mb and tens of genes. To identify the genes most likely involved in cercarial production, we focus on few criteria:\n",
    "* We refine the QTL regions by determining the chromosomal positions corresponding to a drop of 1.8 LOD from the highest LOD score of each QTL. We expect the genes involved to be relatively close to the peaks.\n",
    "* We focus on alternative fix alleles in the parents (with a GQ greater than 29 and a DP greater than 5). The phenotype in the parental population being dramatic, we expect to the parents to be homozygous for the alleles involved.\n",
    "* We take in account mutations and their impacts in each gene. We expect at least on the parental allele to show mutations with important impacts.\n",
    "* We final take in account know gene expression in late sporocyst and cercariae. We expect the genes to be expressed at these stages.\n",
    "\n",
    "Using these criteria, we created a formula to generate a score (genoscore) to objectively prioritize candidate genes in each QTL region:\n",
    "\n",
    "Genoscore = $\\sum_{i=1}^{n} (GSa_i \\times GSb_i \\times IS_i) \\over l \\times \\sqrt(d_peak))$$ \\times e \\times 100$\n",
    "\n",
    "The genoscore is computed either by gene or geneâs coding sequence (CDS). For each site i of a gene or CDS, we first compute a genotype score (GS) for each cross a and b by associating a score of 20 if genotypes are alternatively fixed in each cross parent or a score of 5 if fixed in one parent and unknown in the other (any other genotype combination received 0) and an impact score (IS) from the snpEff annotation (low impact or modifier mutation = 1, moderate impact = 5, high impact = 10). We sum the results for the n sites present in the gene or CDS. We weigh this sum using the length l of the gene or the CDS, and the square root of the distance d between the position of the highest LOD score (peak) and the start of the gene or the CDS. This was finally modulated by an expression factor e of 1 if expressed or -1 otherwise. Jupyter notebook and scripts used to obtain gene expression are available on [Github](https://github.com/fdchevalier/Sm_gene_expression)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Rscript scripts/Candidate_genes.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy number variation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in $(find data/libraries/1-GWAS/ -name \"*_sorted_MD_recal.bam\" | sort)\n",
    "do\n",
    "    sample=$(basename $(dirname \"$i\"))\n",
    "    output=\"$(dirname \"$i\")/${sample}_3_QTL.cov\"\n",
    "    echo \"samtools view -u $i SM_V7_3 | bedtools genomecov -ibam - -d | grep -w SM_V7_3 | awk '\\$3 > 0 {print}' > $output\" | qsub -V -cwd -o \"$statdir\" -j y -r y -pe smp 12\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-06T21:51:44.055141Z",
     "start_time": "2021-06-06T21:51:43.057003Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "bash: qsub: command not found...\n",
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in $(find data/libraries/2-ER-ES_populations/ -name \"*_sorted_MD_recal.bam\" | sort)\n",
    "do\n",
    "    sample=$(basename $(dirname \"$i\"))\n",
    "    output=\"$(dirname \"$i\")/${sample}_3_QTL.cov\"\n",
    "    echo \"samtools view -u $i SM_V7_3 | bedtools genomecov -ibam - -d | grep -w SM_V7_3 | awk '\\$3 > 0 {print}' > $output\" | qsub -V -cwd -o \"$statdir\" -j y -r y -pe smp 12\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SmTRP mutations\n",
    "\n",
    "### GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-06T22:42:46.705573Z",
     "start_time": "2021-06-06T22:42:46.378746Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in $(find data/libraries/1-GWAS -maxdepth 1 -printf '%P\\n' | sed \"/^$/d\")\n",
    "do\n",
    "    echo -e \"$i\\t$(cut -d \"_\" -f 3 <<< \"$i\")\"\n",
    "done | sort -k2 > data/pop_GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-06T22:40:42.094124Z",
     "start_time": "2021-06-06T22:40:41.881934Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "gdir=\"data/genome\"\n",
    "cdir=\"data/calling\"\n",
    "myvcf=\"$cdir/$(grep -m 1 VCF_PFX GWAS.smk | cut -d \"\\\"\" -f 2).vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-06T22:38:57.743692Z",
     "start_time": "2021-06-06T22:38:57.575334Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mutdir=\"results/4-Mutations in SmTRP/1-GWAS\"\n",
    "[[ ! -d \"$mutdir\" ]] && mkdir -p \"$mutdir\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-06T22:40:46.383084Z",
     "start_time": "2021-06-06T22:40:46.025796Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scripts/func-table-mut.sh: line 246: trap: in: invalid signal specification\n",
      "scripts/func-table-mut.sh: line 246: trap: SmTRP/Smp_246790.tsv results/4-Mutations: invalid signal specification\n",
      "scripts/func-table-mut.sh: line 246: trap: in: invalid signal specification\n",
      "scripts/func-table-mut.sh: line 246: trap: SmTRP/[na]_Smp_246790_cds.fa results/4-Mutations: invalid signal specification\n",
      "scripts/func-table-mut.sh: line 246: trap: in: invalid signal specification\n",
      "scripts/func-table-mut.sh: line 246: trap: SmTRP/[aa]_Smp_246790.fa: invalid signal specification\n",
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "scripts/func-table-mut.sh -v \"$myvcf\" \\\n",
    "    -r \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\"\\\n",
    "    -g \"$gdir/Smp_246790.5.gff\" \\\n",
    "    -p data/pop_GWAS \\\n",
    "    -o \"$mutdir/Smp_246790.tsv\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Selected populations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-07T02:14:04.587585Z",
     "start_time": "2021-06-07T02:14:04.278388Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in $(find data/libraries/2-ER-ES_populations -maxdepth 1 -printf '%P\\n' | sed \"/^$/d\")\n",
    "do\n",
    "    echo -e \"$i\\t$(cut -d \"_\" -f 1 <<< \"$i\")\"\n",
    "done | sort -k2 > data/pop_ER-ES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-07T02:14:37.206658Z",
     "start_time": "2021-06-07T02:14:36.981201Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "gdir=\"data/genome\"\n",
    "cdir=\"data/calling\"\n",
    "myvcf=\"$cdir/$(grep -m 1 VCF_PFX Selected_populations.smk | cut -d \"\\\"\" -f 2).vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-07T02:14:45.215359Z",
     "start_time": "2021-06-07T02:14:45.049362Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(PZQ-R_DNA) (PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mutdir=\"results/4-Mutations in SmTRP/2-ER-ES_populations\"\n",
    "[[ ! -d \"$mutdir\" ]] && mkdir -p \"$mutdir\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-07T02:17:07.682499Z",
     "start_time": "2021-06-07T02:14:46.775058Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mInfo:\u001b[00m Identifying samples related to SmLE-PZQ-ER, SmLE-PZQ-ES population(s)\n",
      "\u001b[32mInfo:\u001b[00m Extraction information from GFF\n",
      "\u001b[32mInfo:\u001b[00m Generating reference sequence\n",
      "sed: couldn't write 518 items to stdout: Broken pipe\n",
      "\u001b[32mProgress:\u001b[00m [========================================] 100%\n",
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "scripts/func-table-mut.sh -v \"$myvcf\" \\\n",
    "    -r \"$gdir/schistosoma_mansoni.PRJEA36577.WBPS14.genomic.fa\"\\\n",
    "    -g \"$gdir/Smp_246790.5.gff\" \\\n",
    "    -p data/pop_ER-ES \\\n",
    "    -o \"$mutdir/Smp_246790.5.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-06-07T02:14:40.933087Z",
     "start_time": "2021-06-07T02:14:40.828362Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data/calling/PZQ_ER-ES.vcf.gz\n",
      "(PZQ-R_DNA) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "echo $myvcf"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "228px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
